diff --git a/node_modules/@playwright/test/lib/transform.js b/node_modules/@playwright/test/lib/transform.js
index 7b845a5..b63f6cc 100644
--- a/node_modules/@playwright/test/lib/transform.js
+++ b/node_modules/@playwright/test/lib/transform.js
@@ -90,7 +90,7 @@ function installTransform() {
       presets: [[require.resolve('@babel/preset-typescript'), {
         onlyRemoveTypeImports: true
       }]],
-      plugins: [[require.resolve('@babel/plugin-proposal-class-properties')], [require.resolve('@babel/plugin-proposal-numeric-separator')], [require.resolve('@babel/plugin-proposal-logical-assignment-operators')], [require.resolve('@babel/plugin-proposal-nullish-coalescing-operator')], [require.resolve('@babel/plugin-proposal-optional-chaining')], [require.resolve('@babel/plugin-syntax-json-strings')], [require.resolve('@babel/plugin-syntax-optional-catch-binding')], [require.resolve('@babel/plugin-syntax-async-generators')], [require.resolve('@babel/plugin-syntax-object-rest-spread')], [require.resolve('@babel/plugin-proposal-export-namespace-from')], [require.resolve('@babel/plugin-transform-modules-commonjs')], [require.resolve('@babel/plugin-proposal-dynamic-import')]],
+      plugins: [[require.resolve('@babel/plugin-syntax-import-assertions')],[require.resolve('@babel/plugin-proposal-class-properties')], [require.resolve('@babel/plugin-proposal-numeric-separator')], [require.resolve('@babel/plugin-proposal-logical-assignment-operators')], [require.resolve('@babel/plugin-proposal-nullish-coalescing-operator')], [require.resolve('@babel/plugin-proposal-optional-chaining')], [require.resolve('@babel/plugin-syntax-json-strings')], [require.resolve('@babel/plugin-syntax-optional-catch-binding')], [require.resolve('@babel/plugin-syntax-async-generators')], [require.resolve('@babel/plugin-syntax-object-rest-spread')], [require.resolve('@babel/plugin-proposal-export-namespace-from')], [require.resolve('@babel/plugin-transform-modules-commonjs')], [require.resolve('@babel/plugin-proposal-dynamic-import')]],
       sourceMaps: 'both'
     });
 
