diff --git a/node_modules/domino/lib/defineElement.js b/node_modules/domino/lib/defineElement.js
index 0fd0f19..b0e970c 100644
--- a/node_modules/domino/lib/defineElement.js
+++ b/node_modules/domino/lib/defineElement.js
@@ -23,7 +23,7 @@ module.exports = function(spec, defaultConstructor, tagList, tagNameToImpl) {
     if (spec.events) {
       addEventHandlers(c, spec.events);
     }
-    tagList[c.name] = c;
+    tagList[spec.name] = c;
   }
   else {
     c = defaultConstructor;
diff --git a/node_modules/domino/lib/htmlelts.js b/node_modules/domino/lib/htmlelts.js
index 697b5c8..07a7cae 100644
--- a/node_modules/domino/lib/htmlelts.js
+++ b/node_modules/domino/lib/htmlelts.js
@@ -72,6 +72,7 @@ var HTMLFormElement = function(doc, localName, prefix) {
 
 var HTMLElement = exports.HTMLElement = define({
   superclass: Element,
+  name: 'HTMLElement',
   ctor: function HTMLElement(doc, localName, prefix) {
     Element.call(this, doc, localName, utils.NAMESPACE.HTML, prefix);
   },
@@ -202,6 +203,7 @@ var HTMLElement = exports.HTMLElement = define({
 
 
 var HTMLUnknownElement = define({
+  name: 'HTMLUnknownElement',
   ctor: function HTMLUnknownElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   }
@@ -217,6 +219,7 @@ var formAssociatedProps = {
 
 define({
   tag: 'a',
+  name: 'HTMLAnchorElement',
   ctor: function HTMLAnchorElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -253,6 +256,7 @@ URLUtils._inherit(htmlNameToImpl.a.prototype);
 
 define({
   tag: 'area',
+  name: 'HTMLAreaElement',
   ctor: function HTMLAreaElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -279,6 +283,7 @@ URLUtils._inherit(htmlNameToImpl.area.prototype);
 
 define({
   tag: 'br',
+  name: 'HTMLBRElement',
   ctor: function HTMLBRElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -290,6 +295,7 @@ define({
 
 define({
   tag: 'base',
+  name: 'HTMLBaseElement',
   ctor: function HTMLBaseElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -301,6 +307,7 @@ define({
 
 define({
   tag: 'body',
+  name: 'HTMLBodyElement',
   ctor: function HTMLBodyElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -330,6 +337,7 @@ define({
 
 define({
   tag: 'button',
+  name: 'HTMLButtonElement',
   ctor: function HTMLButtonElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -349,6 +357,7 @@ define({
 
 define({
   tag: 'dl',
+  name: 'HTMLDListElement',
   ctor: function HTMLDListElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -360,6 +369,7 @@ define({
 
 define({
   tag: 'data',
+  name: 'HTMLDataElement',
   ctor: function HTMLDataElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -370,6 +380,7 @@ define({
 
 define({
   tag: 'datalist',
+  name: 'HTMLDataListElement',
   ctor: function HTMLDataListElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   }
@@ -377,6 +388,7 @@ define({
 
 define({
   tag: 'details',
+  name: 'HTMLDetailsElement',
   ctor: function HTMLDetailsElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -387,6 +399,7 @@ define({
 
 define({
   tag: 'div',
+  name: 'HTMLDivElement',
   ctor: function HTMLDivElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -398,6 +411,7 @@ define({
 
 define({
   tag: 'embed',
+  name: 'HTMLEmbedElement',
   ctor: function HTMLEmbedElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -414,6 +428,7 @@ define({
 
 define({
   tag: 'fieldset',
+  name: 'HTMLFieldSetElement',
   ctor: function HTMLFieldSetElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -426,6 +441,7 @@ define({
 
 define({
   tag: 'form',
+  name: 'HTMLFormElement',
   ctor: function HTMLFormElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -445,6 +461,7 @@ define({
 
 define({
   tag: 'hr',
+  name: 'HTMLHRElement',
   ctor: function HTMLHRElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -460,6 +477,7 @@ define({
 
 define({
   tag: 'head',
+  name: 'HTMLHeadElement',
   ctor: function HTMLHeadElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   }
@@ -467,6 +485,7 @@ define({
 
 define({
   tags: ['h1','h2','h3','h4','h5','h6'],
+  name: 'HTMLHeadingElement',
   ctor: function HTMLHeadingElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -478,6 +497,7 @@ define({
 
 define({
   tag: 'html',
+  name: 'HTMLHtmlElement',
   ctor: function HTMLHtmlElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -489,6 +509,7 @@ define({
 
 define({
   tag: 'iframe',
+  name: 'HTMLIFrameElement',
   ctor: function HTMLIFrameElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
     var Window = require('./Window'); // Avoid circular dependencies.
@@ -526,6 +547,7 @@ define({
 
 define({
   tag: 'img',
+  name: 'HTMLImageElement',
   ctor: function HTMLImageElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -552,6 +574,7 @@ define({
 
 define({
   tag: 'input',
+  name: 'HTMLInputElement',
   ctor: function HTMLInputElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -615,6 +638,7 @@ define({
 
 define({
   tag: 'keygen',
+  name: 'HTMLKeygenElement',
   ctor: function HTMLKeygenElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -630,6 +654,7 @@ define({
 
 define({
   tag: 'li',
+  name: 'HTMLLIElement',
   ctor: function HTMLLIElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -642,6 +667,7 @@ define({
 
 define({
   tag: 'label',
+  name: 'HTMLLabelElement',
   ctor: function HTMLLabelElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -653,6 +679,7 @@ define({
 
 define({
   tag: 'legend',
+  name: 'HTMLLegendElement',
   ctor: function HTMLLegendElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -664,6 +691,7 @@ define({
 
 define({
   tag: 'link',
+  name: 'HTMLLinkElement',
   ctor: function HTMLLinkElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -687,6 +715,7 @@ define({
 
 define({
   tag: 'map',
+  name: 'HTMLMapElement',
   ctor: function HTMLMapElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -697,6 +726,7 @@ define({
 
 define({
   tag: 'menu',
+  name: 'HTMLMenuElement',
   ctor: function HTMLMenuElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -712,6 +742,7 @@ define({
 
 define({
   tag: 'meta',
+  name: 'HTMLMetaElement',
   ctor: function HTMLMetaElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -726,6 +757,7 @@ define({
 
 define({
   tag: 'meter',
+  name: 'HTMLMeterElement',
   ctor: function HTMLMeterElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -734,6 +766,7 @@ define({
 
 define({
   tags: ['ins', 'del'],
+  name: 'HTMLModElement',
   ctor: function HTMLModElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -745,6 +778,7 @@ define({
 
 define({
   tag: 'ol',
+  name: 'HTMLOListElement',
   ctor: function HTMLOListElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -781,6 +815,7 @@ define({
 
 define({
   tag: 'object',
+  name: 'HTMLObjectElement',
   ctor: function HTMLObjectElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -809,6 +844,7 @@ define({
 
 define({
   tag: 'optgroup',
+  name: 'HTMLOptGroupElement',
   ctor: function HTMLOptGroupElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -820,6 +856,7 @@ define({
 
 define({
   tag: 'option',
+  name: 'HTMLOptionElement',
   ctor: function HTMLOptionElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -853,6 +890,7 @@ define({
 
 define({
   tag: 'output',
+  name: 'HTMLOutputElement',
   ctor: function HTMLOutputElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -865,6 +903,7 @@ define({
 
 define({
   tag: 'p',
+  name: 'HTMLParagraphElement',
   ctor: function HTMLParagraphElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -876,6 +915,7 @@ define({
 
 define({
   tag: 'param',
+  name: 'HTMLParamElement',
   ctor: function HTMLParamElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -890,6 +930,7 @@ define({
 
 define({
   tags: ['pre',/*legacy elements:*/'listing','xmp'],
+  name: 'HTMLPreElement',
   ctor: function HTMLPreElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -901,6 +942,7 @@ define({
 
 define({
   tag: 'progress',
+  name: 'HTMLProgressElement',
   ctor: function HTMLProgressElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -912,6 +954,7 @@ define({
 
 define({
   tags: ['q', 'blockquote'],
+  name: 'HTMLQuoteElement',
   ctor: function HTMLQuoteElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -922,6 +965,7 @@ define({
 
 define({
   tag: 'script',
+  name: 'HTMLScriptElement',
   ctor: function HTMLScriptElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -958,6 +1002,7 @@ define({
 
 define({
   tag: 'select',
+  name: 'HTMLSelectElement',
   ctor: function HTMLSelectElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -980,6 +1025,7 @@ define({
 
 define({
   tag: 'source',
+  name: 'HTMLSourceElement',
   ctor: function HTMLSourceElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -992,6 +1038,7 @@ define({
 
 define({
   tag: 'span',
+  name: 'HTMLSpanElement',
   ctor: function HTMLSpanElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   }
@@ -999,6 +1046,7 @@ define({
 
 define({
   tag: 'style',
+  name: 'HTMLStyleElement',
   ctor: function HTMLStyleElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1011,6 +1059,7 @@ define({
 
 define({
   tag: 'caption',
+  name: 'HTMLTableCaptionElement',
   ctor: function HTMLTableCaptionElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1022,6 +1071,7 @@ define({
 
 
 define({
+  name: 'HTMLTableCellElement',
   ctor: function HTMLTableCellElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1046,6 +1096,7 @@ define({
 
 define({
   tags: ['col', 'colgroup'],
+  name: 'HTMLTableColElement',
   ctor: function HTMLTableColElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1062,6 +1113,7 @@ define({
 
 define({
   tag: 'table',
+  name: 'HTMLTableElement',
   ctor: function HTMLTableElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1086,6 +1138,7 @@ define({
 
 define({
   tag: 'template',
+  name: 'HTMLTemplateElement',
   ctor: function HTMLTemplateElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
     this._contentFragment = doc._templateDoc.createDocumentFragment();
@@ -1098,6 +1151,7 @@ define({
 
 define({
   tag: 'tr',
+  name: 'HTMLTableRowElement',
   ctor: function HTMLTableRowElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1118,6 +1172,7 @@ define({
 
 define({
   tags: ['thead', 'tfoot', 'tbody'],
+  name: 'HTMLTableSectionElement',
   ctor: function HTMLTableSectionElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1137,6 +1192,7 @@ define({
 
 define({
   tag: 'textarea',
+  name: 'HTMLTextAreaElement',
   ctor: function HTMLTextAreaElement(doc, localName, prefix) {
     HTMLFormElement.call(this, doc, localName, prefix);
   },
@@ -1179,6 +1235,7 @@ define({
 
 define({
   tag: 'time',
+  name: 'HTMLTimeElement',
   ctor: function HTMLTimeElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1190,6 +1247,7 @@ define({
 
 define({
   tag: 'title',
+  name: 'HTMLTitleElement',
   ctor: function HTMLTitleElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1202,6 +1260,7 @@ define({
 
 define({
   tag: 'ul',
+  name: 'HTMLUListElement',
   ctor: function HTMLUListElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1213,6 +1272,7 @@ define({
 });
 
 define({
+  name: 'HTMLMediaElement',
   ctor: function HTMLMediaElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1231,6 +1291,7 @@ define({
 define({
   tag: 'audio',
   superclass: htmlElements.HTMLMediaElement,
+  name: 'HTMLAudioElement',
   ctor: function HTMLAudioElement(doc, localName, prefix) {
     htmlElements.HTMLMediaElement.call(this, doc, localName, prefix);
   }
@@ -1239,6 +1300,7 @@ define({
 define({
   tag: 'video',
   superclass: htmlElements.HTMLMediaElement,
+  name: 'HTMLVideoElement',
   ctor: function HTMLVideoElement(doc, localName, prefix) {
     htmlElements.HTMLMediaElement.call(this, doc, localName, prefix);
   },
@@ -1252,6 +1314,7 @@ define({
 define({
   tag: 'td',
   superclass: htmlElements.HTMLTableCellElement,
+  name: 'HTMLTableDataCellElement',
   ctor: function HTMLTableDataCellElement(doc, localName, prefix) {
     htmlElements.HTMLTableCellElement.call(this, doc, localName, prefix);
   }
@@ -1260,6 +1323,7 @@ define({
 define({
   tag: 'th',
   superclass: htmlElements.HTMLTableCellElement,
+  name: 'HTMLTableHeaderCellElement',
   ctor: function HTMLTableHeaderCellElement(doc, localName, prefix) {
     htmlElements.HTMLTableCellElement.call(this, doc, localName, prefix);
   },
@@ -1267,6 +1331,7 @@ define({
 
 define({
   tag: 'frameset',
+  name: 'HTMLFrameSetElement',
   ctor: function HTMLFrameSetElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   }
@@ -1274,6 +1339,7 @@ define({
 
 define({
   tag: 'frame',
+  name: 'HTMLFrameElement',
   ctor: function HTMLFrameElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   }
@@ -1281,6 +1347,7 @@ define({
 
 define({
   tag: 'canvas',
+  name: 'HTMLCanvasElement',
   ctor: function HTMLCanvasElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1300,6 +1367,7 @@ define({
 
 define({
   tag: 'dialog',
+  name: 'HTMLDialogElement',
   ctor: function HTMLDialogElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1316,6 +1384,7 @@ define({
 
 define({
   tag: 'menuitem',
+  name: 'HTMLMenuItemElement',
   ctor: function HTMLMenuItemElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1354,6 +1423,7 @@ define({
 
 define({
   tag: 'source',
+  name: 'HTMLSourceElement',
   ctor: function HTMLSourceElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1368,6 +1438,7 @@ define({
 
 define({
   tag: 'track',
+  name: 'HTMLTrackElement',
   ctor: function HTMLTrackElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1391,6 +1462,7 @@ define({
 define({
   // obsolete
   tag: 'font',
+  name: 'HTMLFontElement',
   ctor: function HTMLFontElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
@@ -1404,6 +1476,7 @@ define({
 define({
   // obsolete
   tag: 'dir',
+  name: 'HTMLDirectoryElement',
   ctor: function HTMLDirectoryElement(doc, localName, prefix) {
     HTMLElement.call(this, doc, localName, prefix);
   },
diff --git a/node_modules/domino/lib/svg.js b/node_modules/domino/lib/svg.js
index 7e5bf0d..24aca19 100644
--- a/node_modules/domino/lib/svg.js
+++ b/node_modules/domino/lib/svg.js
@@ -18,6 +18,7 @@ function define(spec) {
 
 var SVGElement = define({
   superclass: Element,
+  name: 'SVGElement',
   ctor: function SVGElement(doc, localName, prefix) {
     Element.call(this, doc, localName, utils.NAMESPACE.SVG, prefix);
   },
@@ -31,6 +32,7 @@ var SVGElement = define({
 });
 
 define({
+  name: 'SVGSVGElement',
   ctor: function SVGSVGElement(doc, localName, prefix) {
     SVGElement.call(this, doc, localName, prefix);
   },
