diff --git a/node_modules/ts-node/dist-raw/node-esm-resolve-implementation.js b/node_modules/ts-node/dist-raw/node-esm-resolve-implementation.js
index 04ea846..1a88606 100644
--- a/node_modules/ts-node/dist-raw/node-esm-resolve-implementation.js
+++ b/node_modules/ts-node/dist-raw/node-esm-resolve-implementation.js
@@ -333,8 +333,9 @@ function resolveExtensions(search) {
 const replacementExtensions = extensions.filter(ext => ['.js', '.jsx', '.ts', '.tsx'].includes(ext));
 
 function resolveReplacementExtensions(search) {
-  if (search.pathname.match(/\.js$/)) {
-    const pathnameWithoutExtension = search.pathname.slice(0, search.pathname.length - 3);
+  const ext = require('path').parse(search.pathname).ext;
+  if (ext !== 'js') {
+    const pathnameWithoutExtension = search.pathname.slice(0, search.pathname.length - ext.length);
     for (let i = 0; i < replacementExtensions.length; i++) {
       const extension = replacementExtensions[i];
       const guess = new URL(search.toString());
diff --git a/node_modules/ts-node/dist-raw/node-options.js b/node_modules/ts-node/dist-raw/node-options.js
index 0602a47..4a58b9f 100644
--- a/node_modules/ts-node/dist-raw/node-options.js
+++ b/node_modules/ts-node/dist-raw/node-options.js
@@ -26,6 +26,7 @@ function parseOptions() {
 
 function parseArgv(argv) {
   return require('arg')({
+    '--experimental-json-modules': Boolean,
     '--preserve-symlinks': Boolean,
     '--preserve-symlinks-main': Boolean,
     '--input-type': String,
@@ -97,5 +98,16 @@ function getOptionValuesFromOtherEnvVars() {
   if(process.env.NODE_PENDING_DEPRECATION === '1') {
     options['--pending-deprecation'] = true;
   }
+  if (process.env.npm_config_node_options) {
+      const errors = [];
+      const envArgv = ParseNodeOptionsEnvVar(process.env.npm_config_node_options, errors);
+      if (errors.length) {
+          throw new Error(String(errors));
+      }
+      Object.assign(
+          options,
+          parseArgv(envArgv),
+      );
+  }
   return options;
 }
