diff --git a/applications/calendar/package.json b/applications/calendar/package.json
index aa924b6f9f..cea92a186e 100644
--- a/applications/calendar/package.json
+++ b/applications/calendar/package.json
@@ -6,7 +6,6 @@
     "author": "",
     "main": "index.js",
     "scripts": {
-        "build:web": "cross-env NODE_ENV=production TS_NODE_PROJECT=\"../../tsconfig.webpack.json\" proton-pack build --webpackOnCaffeine --appMode=sso",
         "check-types": "tsc",
         "i18n:extract:web": "proton-i18n extract",
         "i18n:upgrade": "proton-i18n extract --verbose && proton-i18n crowdin --verbose",

diff --git a/applications/calendar/webpack.config.ts b/applications/calendar/webpack.config.ts
index 729c7fc564..8c21b9143e 100644
--- a/applications/calendar/webpack.config.ts
+++ b/applications/calendar/webpack.config.ts
@@ -1,12 +1,12 @@
 import { Configuration } from 'webpack';

 import getConfig from '@proton/pack/webpack.config';
-import { addDevEntry } from '@proton/pack/webpack/entries';
+import { addLoginEntry } from '@proton/pack/webpack/entries';

 const result = (env: any): Configuration => {
     const config = getConfig(env);
-    if (env.appMode === 'standalone') {
-        addDevEntry(config);
+    if (env.appMode === 'standalone') { // keeping original condition, so login page only included for the "mail" app
+        addLoginEntry(config);
     }
     return config;
 };

