diff --git a/package.json b/package.json
index cb439a93b1..2f7931e464 100644
--- a/package.json
+++ b/package.json
@@ -3,15 +3,11 @@
     "private": true,
     "license": "GPL-3.0",
     "workspaces": [
-        "applications/*",
+        "applications/drive",
         "applications/lumo/src/app/lib/lumo-api-client",
-        "applications/pass-desktop/native",
         "packages/*",
         "packages/wasm/*",
         "packages/pass/docs/starlight",
-        "tests",
-        "tests/mail-renderer",
-        "tests/packages/*",
         "utilities/*",
         "vendor/*/*"
     ],
--

diff --git a/yarn.lock b/yarn.lock
index 2fa43fadf7..f709268556 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -11454,13 +11454,6 @@ __metadata:
   languageName: unknown
   linkType: soft

-"@proton/proton-foundation-search@npm:2.0.0-preview2-job33841442":
-  version: 2.0.0-preview2-job33841442
-  resolution: "@proton/proton-foundation-search@npm:2.0.0-preview2-job33841442"
-  checksum: 10/7c51182275c959fd526f34c160f69af5d115e9e4a390f66984045feca3ef70a36aedfbc40e393dab5552bae69d82fa758de5d880b5902b615af6838c86f7b3e5
-  languageName: node
-  linkType: hard
-
 "@proton/proton-foundation-search@npm:^1.1.2":
   version: 1.1.2
   resolution: "@proton/proton-foundation-search@npm:1.1.2"
@@ -41856,7 +41849,6 @@ __metadata:
     "@proton/pack": "workspace:^"
     "@proton/payments": "workspace:^"
     "@proton/polyfill": "workspace:^"
-    "@proton/proton-foundation-search": "npm:2.0.0-preview2-job33841442"
     "@proton/raw-images": "workspace:^"
     "@proton/react-redux-store": "workspace:^"
     "@proton/redux-shared-store": "workspace:^"
--

diff --git a/applications/drive/package.json b/applications/drive/package.json
index 35b3bfa0f2..c4e9f67131 100644
--- a/applications/drive/package.json
+++ b/applications/drive/package.json
@@ -7,7 +7,6 @@
     "main": "index.ts",
     "scripts": {
         "analyze": "yarn build:web --analyze",
-        "build:web": "cross-env NODE_ENV=production TS_NODE_PROJECT=\"../../tsconfig.webpack.json\" proton-pack build --appMode=sso --handleSupportAndErrors --optimizeAssets --noLogicalScss",
         "check-types": "tsc",
         "i18n:extract:local": "yarn workspace @proton/i18n build && yarn build:web && proton-i18n extract",
         "i18n:extract:web": "proton-i18n extract",
@@ -44,7 +43,6 @@
         "@proton/pack": "workspace:^",
         "@proton/payments": "workspace:^",
         "@proton/polyfill": "workspace:^",
-        "@proton/proton-foundation-search": "2.0.0-preview2-job33841442",
         "@proton/raw-images": "workspace:^",
         "@proton/react-redux-store": "workspace:^",
         "@proton/redux-shared-store": "workspace:^",

diff --git a/packages/drive/modules/flags/internal/useFlagsDriveFoundationSearch.ts b/packages/drive/modules/flags/internal/useFlagsDriveFoundationSearch.ts
index 30244e7176..0621735420 100644
--- a/packages/drive/modules/flags/internal/useFlagsDriveFoundationSearch.ts
+++ b/packages/drive/modules/flags/internal/useFlagsDriveFoundationSearch.ts
@@ -1,5 +1,3 @@
-import { useFlag } from '@proton/unleash/useFlag';
-
 export function useFlagsDriveFoundationSearch() {
-    return useFlag('DriveWebSearchFoundation');
+    return false;
 }
--

diff --git a/applications/drive/src/app/index.tsx b/applications/drive/src/app/index.tsx
index 0a9d3cf8bf..a60e5a30b5 100644
--- a/applications/drive/src/app/index.tsx
+++ b/applications/drive/src/app/index.tsx
@@ -9,6 +9,8 @@ import '@proton/polyfill';
 import App from './App';
 import './style';

+window.sessionStorage.setItem("electron_mail:proton_app_name", "proton-drive");
+
 initDriveWebVitalsReporting(false);

 ReactDOM.render(<App />, document.querySelector('.app-root'));
--

diff --git a/applications/drive/src/.htaccess b/applications/drive/src/.htaccess
index b0b25a5fae..11d4e4af0f 100644
--- a/applications/drive/src/.htaccess
+++ b/applications/drive/src/.htaccess
@@ -40,10 +40,12 @@ ErrorDocument 404 /assets/404.html
 </Files>

 # The download service worker gets chunk hash appended to the end of the filename
+# <electron-mail-mark>
 <Files ~ "^downloadSW">
   Header set Service-Worker-Allowed "/"
   Header set Service-Worker "script"
 </Files>
+# </electron-mail-mark>

 <IfModule mod_deflate.c>
   AddOutputFilter INCLUDES;DEFLATE svg

diff --git a/applications/drive/src/app/modules/fileSaver/download.ts b/applications/drive/src/app/modules/fileSaver/download.ts
index 54148fdd45..3208238f77 100644
--- a/applications/drive/src/app/modules/fileSaver/download.ts
+++ b/applications/drive/src/app/modules/fileSaver/download.ts
@@ -75,8 +75,8 @@ function serviceWorkerKeepAlive() {
 export async function initDownloadSW() {
     // wait Service Worker is registered
     await navigator.serviceWorker.register(
-        /* webpackChunkName: "downloadSW" */
-        new URL('./downloadSW', import.meta.url),
+        /* webpackChunkName: "downloadSW" */ // electron-mail-mark
+        new URL('./downloadSW', import.meta.url), // electron-mail-mark
         {
             scope: `/${stripLeadingAndTrailingSlash(PUBLIC_PATH)}`,
         }
diff --git a/packages/drive-store/store/_downloads/fileSaver/download.ts b/packages/drive-store/store/_downloads/fileSaver/download.ts
index 8fa1b3e64e..acc0b2ec4b 100644
--- a/packages/drive-store/store/_downloads/fileSaver/download.ts
+++ b/packages/drive-store/store/_downloads/fileSaver/download.ts
@@ -58,8 +58,8 @@ function serviceWorkerKeepAlive() {
 export async function initDownloadSW() {
     // wait Service Worker is registered
     await navigator.serviceWorker.register(
-        /* webpackChunkName: "downloadSW" */
-        new URL('./downloadSW', import.meta.url),
+        /* webpackChunkName: "downloadSW" */ // electron-mail-mark
+        new URL('./downloadSW', import.meta.url), // electron-mail-mark
         {
             scope: `/${stripLeadingAndTrailingSlash(PUBLIC_PATH)}`,
         }
--

diff --git a/applications/drive/src/app/legacy/store/_shares/shareUrl.ts b/applications/drive/src/app/legacy/store/_shares/shareUrl.ts
index 2ec6be2a3b..dcad75b593 100644
--- a/applications/drive/src/app/legacy/store/_shares/shareUrl.ts
+++ b/applications/drive/src/app/legacy/store/_shares/shareUrl.ts
@@ -46,5 +46,9 @@ export const getSharedLink = (sharedURL?: {
     const url = sharedURL.publicUrl
         ? replaceLocalURL(sharedURL.publicUrl)
         : getAppHref(`/urls/${sharedURL.token}`, APPS.PROTONDRIVE);
-    return `${url}${generatedPassword !== '' ? `#${generatedPassword}` : ''}`;
+    const resultUrl = `${url}${generatedPassword !== '' ? `#${generatedPassword}` : ''}`;
+    return new URL(
+        new URL(resultUrl).href.substring(new URL(resultUrl).origin.length),
+        "https://drive.proton.me",
+    ).toString();
 };
 diff --git a/packages/drive-store/store/_shares/shareUrl.ts b/packages/drive-store/store/_shares/shareUrl.ts
index 717f393fbe..5b432576aa 100644
--- a/packages/drive-store/store/_shares/shareUrl.ts
+++ b/packages/drive-store/store/_shares/shareUrl.ts
@@ -46,5 +46,9 @@ export const getSharedLink = (sharedURL?: {
     const url = sharedURL.publicUrl
         ? replaceLocalURL(sharedURL.publicUrl)
         : getAppHref(`/urls/${sharedURL.token}`, APPS.PROTONDRIVE);
-    return `${url}${generatedPassword !== '' ? `#${generatedPassword}` : ''}`;
+    const resultUrl = `${url}${generatedPassword !== '' ? `#${generatedPassword}` : ''}`;
+    return new URL(
+        new URL(resultUrl).href.substring(new URL(resultUrl).origin.length),
+        "https://drive.proton.me",
+    ).toString();
 };

diff --git a/applications/drive/src/app/modules/search/internal/mainThread/SearchModule.ts b/applications/drive/src/app/modules/search/internal/mainThread/SearchModule.ts
index e5c1a520a9..31de1eb84b 100644
--- a/applications/drive/src/app/modules/search/internal/mainThread/SearchModule.ts
+++ b/applications/drive/src/app/modules/search/internal/mainThread/SearchModule.ts
@@ -241,45 +241,7 @@ export class SearchModule {
     // TODO: Return a discriminated type instead of true/false to propagate the reason of uncomatibitly
     // TODO: Add some UI to explain better why search is not enabled.
     static isEnvironmentCompatible(): boolean {
-        if (isMobile()) {
-            Logger.info('Bad env: Mobile detected');
-            recordEnvironmentIncompatibilityOnce('mobile');
-            return false;
-        }
-
-        // Old Safari (<17) has several issues.
-        // One: it is throttling a lot. First tens of items are done fast but
-        // after ~ 500 items it goes very slowly and after ~ 2500 items it
-        // basically stops without any progress.
-        // Second: in some cases even if indexing finishes, sometimes search
-        // doesnt work. Probably index is not created correctly. Its just few
-        // reported cases and we haven't found the issue yet.
-        // Because of that, its better to not allow search on Safari at all
-        // until we find some way around it.
-        if (isSafari()) {
-            const browser = getBrowser();
-            if (!browser?.version || !new Version(browser.version).isGreaterThanOrEqual('17')) {
-                Logger.info('Bad env: Obsolete safari unsupported');
-                recordEnvironmentIncompatibilityOnce('safari_too_old');
-                return false;
-            }
-        }
-
-        if (typeof SharedWorker === 'undefined') {
-            Logger.info('Bad env: SharedWorker unsupported');
-            recordEnvironmentIncompatibilityOnce('shared_worker_unsupported');
-            return false;
-        }
-
-        if (typeof indexedDB === 'undefined') {
-            Logger.info('Bad env: IndexedDB unsupported');
-            recordEnvironmentIncompatibilityOnce('indexed_db_unsupported');
-            return false;
-        }
-
-        // TODO: Check for indexedDB real availability by creating/deleting a real dummy DB.
-
-        return true;
+        return false;
     }

     /** @internal Reset singleton state — only for use in tests. */
--
