diff --git a/applications/account/webpack.config.ts b/applications/account/webpack.config.ts
index 47b1d60b28..34c4b88afd 100644
--- a/applications/account/webpack.config.ts
+++ b/applications/account/webpack.config.ts
@@ -7,7 +7,7 @@ import 'webpack-dev-server';

 import getConfig from '@proton/pack/webpack.config';
 import CopyIndexHtmlWebpackPlugin from '@proton/pack/webpack/copy-index-html-webpack-plugin';
-import { addDevEntry, getIndexChunks, getSupportedEntry } from '@proton/pack/webpack/entries';
+import { getIndexChunks, getSupportedEntry } from '@proton/pack/webpack/entries';

 import type { HrefLang } from './pages/interface';
 import { getPages } from './pages/pages';
@@ -59,7 +59,7 @@ const result = async (env: any): Promise<webpack.Configuration> => {

     const { pre, unsupported } = config.entry as any;

-    if (env.appMode === 'standalone') {
+    if (env.appMode === 'bundle') {
         config.entry = {
             pre,
             ['private-index']: [path.resolve('./src/app/private.tsx'), getSupportedEntry()],
@@ -79,7 +79,7 @@ const result = async (env: any): Promise<webpack.Configuration> => {
             })
         );

-        addDevEntry(config);
+        // addDevEntry(config);

         return config;
     }

