diff --git a/applications/account/package.json b/applications/account/package.json
index 3980982181..ea76bb0125 100644
--- a/applications/account/package.json
+++ b/applications/account/package.json
@@ -6,7 +6,6 @@
     "author": "",
     "main": "index.js",
     "scripts": {
-        "build:web": "cross-env NODE_ENV=production TS_NODE_PROJECT=\"../../tsconfig.webpack.json\" proton-pack build --webpackOnCaffeine --appMode=sso --logical",
         "check-types": "tsc",
         "i18n:extract:web": "proton-i18n extract",
         "i18n:upgrade": "proton-i18n extract --verbose && proton-i18n crowdin --verbose",

diff --git a/applications/account/webpack.config.ts b/applications/account/webpack.config.ts
index f231539e57..8ff8ab8b68 100644
--- a/applications/account/webpack.config.ts
+++ b/applications/account/webpack.config.ts
@@ -7,7 +7,7 @@ import 'webpack-dev-server';

 import getConfig from '@proton/pack/webpack.config';
 import CopyIndexHtmlWebpackPlugin from '@proton/pack/webpack/copy-index-html-webpack-plugin';
-import { addDevEntry, getIndexChunks, getSupportedEntry } from '@proton/pack/webpack/entries';
+import { getIndexChunks, getSupportedEntry } from '@proton/pack/webpack/entries';

 import type { HrefLang } from './pages/interface';
 import { getPages } from './pages/pages';
@@ -59,7 +59,7 @@ const result = async (env: any): Promise<webpack.Configuration> => {

     const { pre, unsupported } = config.entry as any;

-    if (env.appMode === 'standalone') {
+    if (env.appMode === 'bundle') {
         config.entry = {
             pre,
             private: [path.resolve('./src/app/private.tsx'), getSupportedEntry()],
@@ -79,7 +79,7 @@ const result = async (env: any): Promise<webpack.Configuration> => {
             })
         );

-        addDevEntry(config);
+        // addLoginEntry(config); // login page supposed to be used in the app only for the entry/mail app

         return config;
     }
