diff --git a/applications/vpn-settings/package.json b/applications/vpn-settings/package.json
index 597d3dc8b8..4dd04ce581 100644
--- a/applications/vpn-settings/package.json
+++ b/applications/vpn-settings/package.json
@@ -6,7 +6,6 @@
     "author": "",
     "main": "index.js",
     "scripts": {
-        "build:web": "cross-env NODE_ENV=production TS_NODE_PROJECT=\"../../tsconfig.webpack.json\" proton-pack build --webpackOnCaffeine --appMode=standalone --logical",
         "check-types": "tsc",
         "i18n:extract:web": "proton-i18n extract",
         "i18n:upgrade": "proton-i18n extract --verbose && proton-i18n crowdin --verbose",

diff --git a/applications/vpn-settings/src/app/MainContainer.tsx b/applications/vpn-settings/src/app/MainContainer.tsx
index 839082af37..1df204206f 100644
--- a/applications/vpn-settings/src/app/MainContainer.tsx
+++ b/applications/vpn-settings/src/app/MainContainer.tsx
@@ -54,6 +54,7 @@ import {
     useToggle,
     useUser,
     useUserSettings,
+    AppsDropdown,
 } from '@proton/components';
 import { CancellationReminderSection } from '@proton/components/containers';
 import TwoFactorSection from '@proton/components/containers/account/TwoFactorSection';
@@ -177,7 +178,7 @@ const MainContainer: FunctionComponent = () => {
     const sidebar = (
         <Sidebar
             app={APPS.PROTONVPN_SETTINGS}
-            appsDropdown={null}
+            appsDropdown={<AppsDropdown app={app} />}
             logo={logo}
             expanded={expanded}
             onToggleExpand={onToggleExpand}

diff --git a/packages/components/components/link/AppLink.tsx b/packages/components/components/link/AppLink.tsx
index 67b2981641..329468a6f3 100644
--- a/packages/components/components/link/AppLink.tsx
+++ b/packages/components/components/link/AppLink.tsx
@@ -44,15 +44,6 @@ const AppLink = ({ to, toApp, reloadDocument, children, ...rest }: AppLinkProps,
                 </a>
             );
         }
-        if (APP_NAME === APPS.PROTONVPN_SETTINGS) {
-            const href = getAppHref(to, targetApp);
-            return (
-                // internal link, trusted
-                <a ref={ref} target="_blank" {...rest} href={href}>
-                    {children}
-                </a>
-            );
-        }
         if (authentication.mode === 'standalone') {
             return (
                 <Tooltip title="Disabled in standalone mode">
