diff --git a/package.json b/package.json
index fcdbe7b557..114868cd3d 100644
--- a/package.json
+++ b/package.json
@@ -3,14 +3,11 @@
     "private": true,
     "license": "GPL-3.0",
     "workspaces": [
-        "applications/*",
-        "applications/pass-desktop/native",
+        "applications/vpn-settings",
+        "applications/account",
         "packages/*",
         "packages/wasm/*",
         "packages/pass/docs/starlight",
-        "tests",
-        "tests/mail-renderer",
-        "tests/packages/*",
         "utilities/*",
         "vendor/*/*"
     ],
--

diff --git a/applications/vpn-settings/package.json b/applications/vpn-settings/package.json
index 712dc84403..69f376912a 100644
--- a/applications/vpn-settings/package.json
+++ b/applications/vpn-settings/package.json
@@ -5,7 +5,6 @@
     "license": "GPL-3.0",
     "author": "",
     "scripts": {
-        "build:web": "cross-env NODE_ENV=production TS_NODE_PROJECT=\"../../tsconfig.webpack.json\" proton-pack build --appMode=standalone",
         "check-types": "tsc",
         "i18n:extract:web": "proton-i18n extract",
         "i18n:upgrade": "proton-i18n extract --verbose && proton-i18n crowdin --verbose",

diff --git a/applications/vpn-settings/webpack.config.ts b/applications/vpn-settings/webpack.config.ts
index 2cd919188c..5ef859b4c9 100644
--- a/applications/vpn-settings/webpack.config.ts
+++ b/applications/vpn-settings/webpack.config.ts
@@ -52,18 +52,10 @@ const result = async (opts: WebpackEnvArgumentsV2): Promise<Configuration> => {
         throw new Error('Missing html plugin');
     }

-    const rewrites: any[] = [];
-    // @ts-ignore
-    config.devServer.historyApiFallback.rewrites = rewrites;
-
     const originalTemplateParameters = htmlPlugin.userOptions.templateParameters as { [key: string]: any };

     const { pages, hreflangs } = await pagesPromise;

-    pages.forEach(({ rewrite }) => {
-        rewrites.push(rewrite);
-    });
-
     plugins.push(
         new CopyIndexHtmlWebpackPlugin((source) => {
             const compiled = template(
--

diff --git a/applications/vpn-settings/src/app/App.tsx b/applications/vpn-settings/src/app/App.tsx
index 7a11c5b624..f4d73e2985 100644
--- a/applications/vpn-settings/src/app/App.tsx
+++ b/applications/vpn-settings/src/app/App.tsx
@@ -5,7 +5,6 @@ import { LocationThemeSync } from 'proton-account/src/app/content/theme/Location
 import { PublicAppThemeProvider } from 'proton-account/src/app/content/theme/PublicAppThemeProvider';

 import * as bootstrap from '@proton/account/bootstrap';
-import { initStandaloneSession } from '@proton/account/bootstrap/standaloneSession';
 import { staticExperimentsConfig } from '@proton/account/staticExperiments/config';
 import { resolveStaticExperiments } from '@proton/account/staticExperiments/resolve';
 import { staticExperimentsActions } from '@proton/account/staticExperiments/slice';
@@ -17,7 +16,6 @@ import AuthenticationProvider from '@proton/components/containers/authentication
 import useInstance from '@proton/hooks/useInstance';
 import { ProtonStoreProvider } from '@proton/redux-shared-store/sharedProvider';
 import createApi from '@proton/shared/lib/api/createApi';
-import type { PersistedSession } from '@proton/shared/lib/authentication/SessionInterface';
 import { handleLogoutFromURL } from '@proton/shared/lib/authentication/handleLogoutFromURL';
 import { getLoginPath } from '@proton/shared/lib/authentication/loginPath';
 import { replaceUrl } from '@proton/shared/lib/helpers/browser';
@@ -25,7 +23,7 @@ import { initSafariFontFixClassnames } from '@proton/shared/lib/helpers/initSafa
 import { telemetry } from '@proton/shared/lib/telemetry';

 import PrivateApp from './PrivateApp';
-import PublicApp, { publicRoutes } from './PublicApp';
+import PublicApp from './PublicApp';
 import config from './config';
 import locales from './locales';
 import { extendStore, setupStore } from './store/store';
@@ -39,20 +37,7 @@ const bootstrapApp = () => {
     bootstrap.init({ config, authentication, locales });
     initSafariFontFixClassnames();
     extendStore({ authentication, api, history, config });
-    let session: PersistedSession | undefined = undefined;
-    // Don't automatically sign in when a public route is opened, for example signup or reset password
-    if (
-        !Object.values(publicRoutes).some((value) => {
-            // Ignore login, let the session get picked automatically
-            if (value === publicRoutes.login) {
-                return false;
-            }
-            return location.pathname.endsWith(value);
-        })
-    ) {
-        session = initStandaloneSession({ authentication, api });
-    }
-    const privateApp = Boolean(session);
+    const privateApp = Boolean(authentication.UID);

     if (!privateApp) {
         telemetry.init({

diff --git a/applications/vpn-settings/src/app/index.tsx b/applications/vpn-settings/src/app/index.tsx
index c75b6faf9a..6b4aecd088 100644
--- a/applications/vpn-settings/src/app/index.tsx
+++ b/applications/vpn-settings/src/app/index.tsx
@@ -5,6 +5,8 @@ import '@proton/polyfill';
 import App from './App';
 import './style';

+window.sessionStorage.setItem("electron_mail:proton_app_name", "proton-vpn-settings");
+
 const container = document.querySelector('.app-root');
 const root = createRoot(container!);
 root.render(<App />);
--

diff --git a/applications/vpn-settings/src/app/VPNSidebar.tsx b/applications/vpn-settings/src/app/VPNSidebar.tsx
index 65f94dd2f8..a775d89300 100644
--- a/applications/vpn-settings/src/app/VPNSidebar.tsx
+++ b/applications/vpn-settings/src/app/VPNSidebar.tsx
@@ -2,6 +2,7 @@ import { useLocation } from 'react-router-dom';

 import Loader from '@proton/components/components/loader/Loader';
 import MainLogo from '@proton/components/components/logo/MainLogo';
+import AppsDropdown from '@proton/components/containers/app/AppsDropdown';
 import SettingsListItem from '@proton/components/components/sidebar/SettingsListItem';
 import Sidebar from '@proton/components/components/sidebar/Sidebar';
 import SidebarList from '@proton/components/components/sidebar/SidebarList';
@@ -33,7 +34,7 @@ export const VPNSidebar = ({ routes, organizationRoutes, sidebarExpanded, onSide
         return (
             <Sidebar
                 app={APPS.PROTONVPN_SETTINGS}
-                appsDropdown={null}
+                appsDropdown={<AppsDropdown app={APPS.PROTONDRIVE} />}
                 logo={<MainLogo to="/" />}
                 expanded={sidebarExpanded}
                 onToggleExpand={onSidebarToggle}
@@ -50,7 +51,7 @@ export const VPNSidebar = ({ routes, organizationRoutes, sidebarExpanded, onSide
     return (
         <Sidebar
             app={APPS.PROTONVPN_SETTINGS}
-            appsDropdown={null}
+            appsDropdown={<AppsDropdown app={APPS.PROTONDRIVE} />}
             logo={<MainLogo to="/" />}
             expanded={sidebarExpanded}
             onToggleExpand={onSidebarToggle}
