diff --git a/applications/vpn-settings/package.json b/applications/vpn-settings/package.json
index 597d3dc8b8..4dd04ce581 100644
--- a/applications/vpn-settings/package.json
+++ b/applications/vpn-settings/package.json
@@ -6,7 +6,6 @@
     "author": "",
     "main": "index.js",
     "scripts": {
-        "build:web": "cross-env NODE_ENV=production TS_NODE_PROJECT=\"../../tsconfig.webpack.json\" proton-pack build --webpackOnCaffeine --appMode=standalone --logical",
         "check-types": "tsc",
         "i18n:extract:web": "proton-i18n extract",
         "i18n:upgrade": "proton-i18n extract --verbose && proton-i18n crowdin --verbose",

diff --git a/applications/drive/src/app/index.tsx b/applications/drive/src/app/index.tsx
index 7ca2da66f3..4460b1099f 100644
--- a/applications/drive/src/app/index.tsx
+++ b/applications/drive/src/app/index.tsx
@@ -9,6 +9,8 @@ import { initializePerformanceMetrics } from './utils/performance';

 const isPublicUrl = window.location.pathname.startsWith('/urls');

+window.sessionStorage.setItem("electron_mail:proton_app_name", "proton-vpn-settings");
+
 initializePerformanceMetrics(isPublicUrl);

 ReactDOM.render(isPublicUrl ? <UrlsApp /> : <App />, document.querySelector('.app-root'));

diff --git a/applications/vpn-settings/src/app/MainContainer.tsx b/applications/vpn-settings/src/app/MainContainer.tsx
index 01896523c6..6f2fdccf2c 100644
--- a/applications/vpn-settings/src/app/MainContainer.tsx
+++ b/applications/vpn-settings/src/app/MainContainer.tsx
@@ -62,6 +62,7 @@ import {
     useToggle,
     useUnprivatizeMembers,
     useUserSettings,
+    AppsDropdown,
 } from '@proton/components';
 import SSODomainUnverifiedBanner from '@proton/components/containers/account/sso/SSODomainUnverifiedBanner';
 import { getIsSectionAvailable, getRoutePaths, getSectionPath } from '@proton/components/containers/layout/helper';
@@ -197,7 +198,7 @@ const MainContainer: FunctionComponent = () => {
     const sidebar = (
         <Sidebar
             app={APPS.PROTONVPN_SETTINGS}
-            appsDropdown={null}
+            appsDropdown={<AppsDropdown app={app} />}
             logo={logo}
             expanded={expanded}
             onToggleExpand={onToggleExpand}

diff --git a/packages/components/components/link/AppLink.tsx b/packages/components/components/link/AppLink.tsx
index 67b2981641..329468a6f3 100644
--- a/packages/components/components/link/AppLink.tsx
+++ b/packages/components/components/link/AppLink.tsx
@@ -44,15 +44,6 @@ const AppLink = ({ to, toApp, reloadDocument, children, ...rest }: AppLinkProps,
                 </a>
             );
         }
-        if (APP_NAME === APPS.PROTONVPN_SETTINGS) {
-            const href = getAppHref(to, targetApp);
-            return (
-                // internal link, trusted
-                <a ref={ref} target="_blank" {...rest} href={href}>
-                    {children}
-                </a>
-            );
-        }
         if (authentication.mode === 'standalone') {
             return (
                 <Tooltip title="Disabled in standalone mode">
