diff --git a/applications/drive/package.json b/applications/drive/package.json
index c037e3ce53..e1655f0a06 100644
--- a/applications/drive/package.json
+++ b/applications/drive/package.json
@@ -7,7 +7,6 @@
     "main": "index.ts",
     "scripts": {
         "analyze": "yarn build:web --analyze",
-        "build:web": "cross-env NODE_ENV=production TS_NODE_PROJECT=\"../../tsconfig.webpack.json\" proton-pack build --webpackOnCaffeine --appMode=sso",
         "check-types": "tsc",
         "i18n:extract:local": "yarn workspace @proton/i18n build && yarn build:web && proton-i18n extract",
         "i18n:extract:web": "proton-i18n extract",

diff --git a/applications/drive/src/app/index.tsx b/applications/drive/src/app/index.tsx
index 7ca2da66f3..2a99b061bb 100644
--- a/applications/drive/src/app/index.tsx
+++ b/applications/drive/src/app/index.tsx
@@ -9,6 +9,8 @@ import { initializePerformanceMetrics } from './utils/performance';

 const isPublicUrl = window.location.pathname.startsWith('/urls');

+window.sessionStorage.setItem("electron_mail:proton_app_name", "proton-drive");
+
 initializePerformanceMetrics(isPublicUrl);

 ReactDOM.render(isPublicUrl ? <UrlsApp /> : <App />, document.querySelector('.app-root'));

diff --git a/applications/drive/src/.htaccess b/applications/drive/src/.htaccess
index b0b25a5fae..11d4e4af0f 100644
--- a/applications/drive/src/.htaccess
+++ b/applications/drive/src/.htaccess
@@ -40,10 +40,12 @@ ErrorDocument 404 /assets/404.html
 </Files>

 # The download service worker gets chunk hash appended to the end of the filename
+# <electron-mail-mark>
 <Files ~ "^downloadSW">
   Header set Service-Worker-Allowed "/"
   Header set Service-Worker "script"
 </Files>
+# </electron-mail-mark>

 <IfModule mod_deflate.c>
   AddOutputFilter INCLUDES;DEFLATE svg

diff --git a/applications/drive/src/app/store/_downloads/fileSaver/download.ts b/applications/drive/src/app/store/_downloads/fileSaver/download.ts
index e3a5bd4889..da2217e2f3 100644
--- a/applications/drive/src/app/store/_downloads/fileSaver/download.ts
+++ b/applications/drive/src/app/store/_downloads/fileSaver/download.ts
@@ -1,6 +1,5 @@
 import { WritableStream } from 'web-streams-polyfill';

-import { isIos, isSafari } from '@proton/shared/lib/helpers/browser';
 import { stripLeadingAndTrailingSlash } from '@proton/shared/lib/helpers/string';
 import { PUBLIC_PATH } from '@proton/shared/lib/webpack.constants';

@@ -15,17 +14,7 @@ let workerWakeupInterval: ReturnType<typeof setInterval>;
  * IOS - forces all browsers to use webkit, so same problems as safari in all browsers.
  * For them download is done in-memory using blob response.
  */
-export const isUnsupported = () => {
-    /* TODO: To be removed after test DRVWEB-4375 */
-    if (typeof window !== 'undefined') {
-        const isSWForSafariEnabled = 'isSWForSafariEnabled' in window && window.isSWForSafariEnabled;
-        if (isSWForSafariEnabled) {
-            return !('serviceWorker' in navigator);
-        }
-    }
-    // Original
-    return !('serviceWorker' in navigator) || isSafari() || isIos();
-};
+export const isUnsupported = () => !('serviceWorker' in navigator);

 // createDownloadIframe opens download URL created in service worker to
 // initialize the download in the browser. The response has headers to
@@ -73,6 +62,7 @@ function serviceWorkerKeepAlive() {
     }, 10000);
 }

+/* <electron-mail-mark> */
 export async function initDownloadSW() {
     if (isUnsupported()) {
         throw new Error('Saving file via download is unsupported by this browser');
@@ -81,7 +71,7 @@ export async function initDownloadSW() {
     await navigator.serviceWorker
         .register(
             /* webpackChunkName: "downloadSW" */
-            new URL('./downloadSW', import.meta.url),
+            new URL('./downloadSW', import.meta.url), // electron-mail-mark
             {
                 scope: `/${stripLeadingAndTrailingSlash(PUBLIC_PATH)}`,
             }
@@ -107,6 +97,7 @@ export async function initDownloadSW() {

     serviceWorkerKeepAlive();
 }
+/* </electron-mail-mark> */

 /**
  * Opens download stream into service worker. Use abort signal when pipeTo can't close the download stream.

diff --git a/applications/drive/src/app/store/_shares/shareUrl.ts b/applications/drive/src/app/store/_shares/shareUrl.ts
index 1b76b8ea81..748ede869b 100644
--- a/applications/drive/src/app/store/_shares/shareUrl.ts
+++ b/applications/drive/src/app/store/_shares/shareUrl.ts
@@ -43,5 +43,9 @@ export const getSharedLink = (sharedURL?: {
     const url = sharedURL.publicUrl
         ? replaceLocalURL(sharedURL.publicUrl)
         : getAppHref(`/urls/${sharedURL.token}`, APPS.PROTONDRIVE);
-    return `${url}${generatedPassword !== '' ? `#${generatedPassword}` : ''}`;
+    const resultUrl = `${url}${generatedPassword !== '' ? `#${generatedPassword}` : ''}`;
+    return new URL(
+        new URL(resultUrl).href.substring(new URL(resultUrl).origin.length),
+        "https://drive.proton.me",
+    ).toString();
 };
--
