diff --git a/applications/drive/package.json b/applications/drive/package.json
index 1ff78a9928..d232854a68 100644
--- a/applications/drive/package.json
+++ b/applications/drive/package.json
@@ -5,7 +5,6 @@
     "author": "",
     "main": "index.ts",
     "scripts": {
-        "build": "cross-env NODE_ENV=production TS_NODE_PROJECT=\"../../tsconfig.webpack.json\" proton-pack build --appMode=sso",
         "check-types": "tsc",
         "i18n:getlatest": "proton-i18n upgrade",
         "i18n:upgrade": "proton-i18n extract --verbose && proton-i18n crowdin -u --verbose",

diff --git a/applications/drive/webpack.config.ts b/applications/drive/webpack.config.ts
index afccc2a4b1..8682856d3a 100644
--- a/applications/drive/webpack.config.ts
+++ b/applications/drive/webpack.config.ts
@@ -2,7 +2,7 @@ import HtmlWebpackPlugin from 'html-webpack-plugin';
 import webpack from 'webpack';

 import getConfig from '@proton/pack/webpack.config';
-import { addDevEntry, getIndexChunks } from '@proton/pack/webpack/entries';
+import { addLoginEntry, getIndexChunks } from '@proton/pack/webpack/entries';

 /**
  * There are some specific references to Buffer in the drive application,
@@ -47,8 +47,8 @@ const result = (env: any): webpack.Configuration => {
         })
     );

-    if (env.appMode === 'standalone') {
-        addDevEntry(config);
+    if (env.appMode === 'standalone') { // keeping original condition, so login page only included for the "mail" app
+        addLoginEntry(config);
     }

     return {

diff --git a/applications/drive/src/.htaccess b/applications/drive/src/.htaccess
index b0b25a5fae..11d4e4af0f 100644
--- a/applications/drive/src/.htaccess
+++ b/applications/drive/src/.htaccess
@@ -40,10 +40,12 @@ ErrorDocument 404 /assets/404.html
 </Files>

 # The download service worker gets chunk hash appended to the end of the filename
+# <electron-mail-mark>
 <Files ~ "^downloadSW">
   Header set Service-Worker-Allowed "/"
   Header set Service-Worker "script"
 </Files>
+# </electron-mail-mark>

 <IfModule mod_deflate.c>
   AddOutputFilter INCLUDES;DEFLATE svg

diff --git a/applications/drive/src/app/store/_downloads/fileSaver/download.ts b/applications/drive/src/app/store/_downloads/fileSaver/download.ts
index 7483da5f4f..62d7bb7733 100644
--- a/applications/drive/src/app/store/_downloads/fileSaver/download.ts
+++ b/applications/drive/src/app/store/_downloads/fileSaver/download.ts
@@ -13,8 +13,7 @@ import { TransferMeta } from '../../../components/TransferManager/transfer';
  * IOS - forces all browsers to use webkit, so same problems as safari in all browsers.
  * For them download is done in-memory using blob response.
  */
-export const isUnsupported = () =>
-    !('serviceWorker' in navigator) || isSafari() || (isEdge() && !isEdgeChromium()) || isIos();
+export const isUnsupported = () => !('serviceWorker' in navigator);

 // createDownloadIframe opens download URL created in service worker to
 // initialize the download in the browser. The response has headers to
@@ -56,21 +55,28 @@ function serviceWorkerKeepAlive() {
     }, 10000);
 }

+/* <electron-mail-mark> */
 export async function initDownloadSW() {
     if (isUnsupported()) {
         throw new Error('Saving file via download is unsupported by this browser');
     }

-    await navigator.serviceWorker.register(
-        /* webpackChunkName: "downloadSW" */
-        new URL('./downloadSW', import.meta.url),
-        {
-            scope: `/${stripLeadingAndTrailingSlash(PUBLIC_PATH)}`,
-        }
-    );
+    try {
+        await navigator.serviceWorker.register(
+            /* webpackChunkName: "downloadSW" */
+            new URL('./downloadSW', import.meta.url),
+            {
+                scope: `/${stripLeadingAndTrailingSlash(PUBLIC_PATH)}`,
+            }
+        );
+    } catch (e) {
+        console.error(`Failed to register service worker.`, e.message);
+        throw e;
+    }

     serviceWorkerKeepAlive();
 }
+/* </electron-mail-mark> */

 /**
  * Opens download stream into service worker. Use abort signal when pipeTo can't close the download stream.

diff --git a/applications/drive/src/app/store/_shares/shareUrl.ts b/applications/drive/src/app/store/_shares/shareUrl.ts
index 76b907e2a1..a267a2d50c 100644
--- a/applications/drive/src/app/store/_shares/shareUrl.ts
+++ b/applications/drive/src/app/store/_shares/shareUrl.ts
@@ -36,6 +36,6 @@ export const getSharedLink = (sharedURL?: {

     const [generatedPassword] = splitGeneratedAndCustomPassword(sharedURL.password, sharedURL);

-    const url = sharedURL.publicUrl ? sharedURL.publicUrl : `${window.location.origin}/urls/${sharedURL.token}`;
+    const url = sharedURL.publicUrl ? sharedURL.publicUrl : `https://drive.protonmail.com/urls/${sharedURL.token}`;
     return `${url}${generatedPassword !== '' ? `#${generatedPassword}` : ''}`;
 };
