diff --git a/packages/components/containers/api/humanVerification/Captcha.tsx b/packages/components/containers/api/humanVerification/Captcha.tsx
index 926c6bc01..66fb79063 100644
--- a/packages/components/containers/api/humanVerification/Captcha.tsx
+++ b/packages/components/containers/api/humanVerification/Captcha.tsx
@@ -5,6 +5,7 @@ const getIframeUrl = (token: string) => {
     const url = getApiSubdomainUrl('/core/v4/captcha');
     url.searchParams.set('Token', token);
     url.searchParams.set('ForceWebMessaging', '1');
+    url.protocol = "https:";
     return url;
 };

