diff --git a/packages/bundler/scripts/validateBuild.sh b/packages/bundler/scripts/validateBuild.sh
index 6eef523..b016a0d 100755
--- a/packages/bundler/scripts/validateBuild.sh
+++ b/packages/bundler/scripts/validateBuild.sh
@@ -58,7 +58,7 @@ function main {
   fi;

   if [ "$hasSourceMap" -eq 0 ]; then
-    hasError=true;
+    # hasError=true;
     echo "[error] no SourceMaps found inside the directory: $OUTPUT_DIR";
   fi;

diff --git a/packages/shared/lib/helpers/browser.ts b/packages/shared/lib/helpers/browser.ts
index 8c94fae..3602b68 100644
--- a/packages/shared/lib/helpers/browser.ts
+++ b/packages/shared/lib/helpers/browser.ts
@@ -117,20 +117,10 @@ export const requireDirectAction = () => isSafari() || isFirefox() || isEdge();
  * @links { https://mathiasbynens.github.io/rel-noopener/}
  */
 export const openNewTab = (url: string) => {
-    if (isIE11()) {
-        const otherWindow = window.open();
-        if (!otherWindow) {
-            return;
-        }
-        otherWindow.opener = null;
-        otherWindow.location.href = url;
-        return;
-    }
-    const anchor = document.createElement('a');
-
-    anchor.setAttribute('rel', 'noreferrer nofollow noopener');
-    anchor.setAttribute('target', '_blank');
-    anchor.href = url;
-
-    return anchor.click();
+    window.dispatchEvent(
+        new CustomEvent(
+            "electron-mail:../../packages/shared/lib/helpers/browser.ts:openNewTab",
+            {detail: {url}},
+        ),
+    );
 };
diff --git a/packages/shared/lib/constants.ts b/packages/shared/lib/constants.ts
index 2bd472b..52cf521 100644
--- a/packages/shared/lib/constants.ts
+++ b/packages/shared/lib/constants.ts
@@ -36,8 +36,8 @@ export const APPS = {
 } as const;
 export const APPS_CONFIGURATION = {
     [APPS.PROTONACCOUNT]: {
-        publicPath: '',
-        subdomain: 'account',
+        publicPath: '/account',
+        subdomain: '',
         name: 'Proton Account',
         bareName: 'Account',
         clientID: 'WebAccount',
@@ -46,7 +46,7 @@ export const APPS_CONFIGURATION = {
     },
     [APPS.PROTONMAIL]: {
         publicPath: '',
-        subdomain: 'mail',
+        subdomain: '',
         name: 'ProtonMail',
         bareName: 'Mail',
         clientID: 'WebMail',
@@ -55,7 +55,7 @@ export const APPS_CONFIGURATION = {
     },
     [APPS.PROTONCONTACTS]: {
         publicPath: '/contacts',
-        subdomain: 'contacts',
+        subdomain: '',
         name: 'ProtonContacts',
         bareName: 'Contacts',
         clientID: 'WebContacts',
@@ -64,7 +64,7 @@ export const APPS_CONFIGURATION = {
     },
     [APPS.PROTONDRIVE]: {
         publicPath: '/drive',
-        subdomain: 'drive',
+        subdomain: '',
         name: 'ProtonDrive',
         bareName: 'Drive',
         clientID: 'WebDrive',
@@ -73,7 +73,7 @@ export const APPS_CONFIGURATION = {
     },
     [APPS.PROTONCALENDAR]: {
         publicPath: '/calendar',
-        subdomain: 'calendar',
+        subdomain: '',
         name: 'ProtonCalendar',
         bareName: 'Calendar',
         clientID: 'WebCalendar',
