diff --git a/packages/shared/lib/constants.ts b/packages/shared/lib/constants.ts
index e3af4c1dc..aff74c45f 100644
--- a/packages/shared/lib/constants.ts
+++ b/packages/shared/lib/constants.ts
@@ -44,8 +44,8 @@ export const APPS = {
 } as const;
 export const APPS_CONFIGURATION = {
     [APPS.PROTONACCOUNT]: {
-        publicPath: '',
-        subdomain: 'account',
+        publicPath: '/account',
+        subdomain: '',
         name: ACCOUNT_APP_NAME,
         bareName: 'Account',
         clientID: 'web-account',
@@ -63,7 +63,7 @@ export const APPS_CONFIGURATION = {
     },
     [APPS.PROTONMAIL]: {
         publicPath: '',
-        subdomain: 'mail',
+        subdomain: '',
         name: MAIL_APP_NAME,
         bareName: 'Mail',
         clientID: 'web-mail',
@@ -72,7 +72,7 @@ export const APPS_CONFIGURATION = {
     },
     [APPS.PROTONCONTACTS]: {
         publicPath: '/contacts',
-        subdomain: 'contacts',
+        subdomain: '',
         name: 'ProtonContacts',
         bareName: 'Contacts',
         clientID: 'web-contacts',
@@ -81,7 +81,7 @@ export const APPS_CONFIGURATION = {
     },
     [APPS.PROTONDRIVE]: {
         publicPath: '/drive',
-        subdomain: 'drive',
+        subdomain: '',
         name: DRIVE_APP_NAME,
         bareName: 'Drive',
         clientID: 'web-drive',
@@ -90,7 +90,7 @@ export const APPS_CONFIGURATION = {
     },
     [APPS.PROTONCALENDAR]: {
         publicPath: '/calendar',
-        subdomain: 'calendar',
+        subdomain: '',
         name: CALENDAR_APP_NAME,
         bareName: 'Calendar',
         clientID: 'web-calendar',
@@ -98,7 +98,7 @@ export const APPS_CONFIGURATION = {
         settingsSlug: 'calendar',
     },
     [APPS.PROTONVPN_SETTINGS]: {
-        publicPath: '',
+        publicPath: 'account/vpn',
         subdomain: '',
         name: VPN_APP_NAME,
         bareName: 'VPN',

