diff --git a/packages/shared/lib/constants.ts b/packages/shared/lib/constants.ts
index d035e6c88..96af3775b 100644
--- a/packages/shared/lib/constants.ts
+++ b/packages/shared/lib/constants.ts
@@ -69,7 +69,7 @@ interface AppConfiguration {

 export const APPS_CONFIGURATION: { [key in APP_NAMES]: AppConfiguration } = {
     [APPS.PROTONACCOUNT]: {
-        publicPath: '',
+        publicPath: '/account',
         subdomain: 'account',
         name: 'Proton Account',
         bareName: 'Account',
@@ -141,8 +141,8 @@ export const APPS_CONFIGURATION: { [key in APP_NAMES]: AppConfiguration } = {
         settingsSlug: '',
     },
     [APPS.PROTONVPN_SETTINGS]: {
-        publicPath: '',
-        subdomain: '',
+        publicPath: 'account/vpn',
+        subdomain: 'account',
         name: VPN_APP_NAME,
         bareName: VPN_SHORT_APP_NAME,
         clientID: 'web-vpn-settings',
