diff --git a/packages/components/hooks/usePendingUserInvitations.ts b/packages/components/hooks/usePendingUserInvitations.ts
index d61e10f07..6cc0d2097 100644
--- a/packages/components/hooks/usePendingUserInvitations.ts
+++ b/packages/components/hooks/usePendingUserInvitations.ts
@@ -1,7 +1,7 @@
 import { useCallback } from 'react';

 import { PendingInvitation as PendingUserInvitation } from '@proton/shared/lib/interfaces';
-import { UserInvitationModel } from '@proton/shared/lib/models';
+import { UserInvitationModelKey } from '@proton/shared/lib/models/userInvitationModel.key';
 import { fetchPendingUserInvitations } from '@proton/shared/lib/models/userInvitationModelApi';

 import useApi from './useApi';
@@ -13,7 +13,7 @@ const usePendingUserInvitations = (): [PendingUserInvitation[] | undefined, bool
     const cache = useCache();

     const miss = useCallback(() => fetchPendingUserInvitations(api), [api]);
-    return useCachedModelResult(cache, UserInvitationModel.key, miss);
+    return useCachedModelResult(cache, UserInvitationModelKey, miss);
 };

 export default usePendingUserInvitations;
diff --git a/packages/shared/lib/models/userInvitationModel.js b/packages/shared/lib/models/userInvitationModel.js
index ec3e45c0e..c8b5672ed 100644
--- a/packages/shared/lib/models/userInvitationModel.js
+++ b/packages/shared/lib/models/userInvitationModel.js
@@ -1,8 +1,9 @@
 import updateCollection from '../helpers/updateCollection';
 import { fetchPendingUserInvitations } from './userInvitationModelApi';
+import { UserInvitationModelKey as key } from './userInvitationModel.key';

 export const UserInvitationModel = {
-    key: 'UserInvitations',
+    key,
     get: fetchPendingUserInvitations,
     update: (model, events) => updateCollection({ model, events, itemKey: 'UserInvitation' }),
 };
diff --git a/packages/shared/lib/models/userInvitationModel.key.js b/packages/shared/lib/models/userInvitationModel.key.js
new file mode 100644
index 000000000..c07ff03fa
--- /dev/null
+++ b/packages/shared/lib/models/userInvitationModel.key.js
@@ -0,0 +1 @@
+export const UserInvitationModelKey = 'UserInvitations';
