diff --git a/packages/components/containers/index.ts b/packages/components/containers/index.ts
index ef8e3cabce..c9d227dad2 100644
--- a/packages/components/containers/index.ts
+++ b/packages/components/containers/index.ts
@@ -64,5 +64,4 @@ export * from './support';
 export * from './themes';
 export * from './topBanners';
 export * from './unleash';
-export * from './verification';
 export * from './vpn';

diff --git a/packages/components/containers/verification/EmbeddedVerification.tsx b/packages/components/containers/verification/EmbeddedVerification.tsx
deleted file mode 100644
index 6425df9ec..000000000
--- a/packages/components/containers/verification/EmbeddedVerification.tsx
+++ /dev/null
@@ -1,103 +0,0 @@
-import { useEffect, useMemo, useRef, useState } from 'react';
-
-import { APPS } from '@proton/shared/lib/constants';
-import { getAppUrlFromApiUrl, getAppUrlRelativeToOrigin, stringifySearchParams } from '@proton/shared/lib/helpers/url';
-import { localeCode } from '@proton/shared/lib/i18n';
-import { HumanVerificationMethodType } from '@proton/shared/lib/interfaces';
-
-import { useConfig, useNotifications } from '../../hooks';
-
-interface EmbeddedVerificationProps {
-    token: string;
-    methods: HumanVerificationMethodType[];
-    defaultCountry?: string;
-    defaultEmail?: string;
-    defaultPhone?: string;
-    onSuccess: (token: string, tokenType: HumanVerificationMethodType) => void;
-}
-
-const EmbeddedVerification = ({
-    token,
-    methods,
-    onSuccess,
-    defaultCountry,
-    defaultEmail,
-    defaultPhone,
-}: EmbeddedVerificationProps) => {
-    const iframeRef = useRef<HTMLIFrameElement>(null);
-    const [iframeHeight, setIframeHeight] = useState<number | undefined>();
-    const { API_URL } = useConfig();
-    const { createNotification } = useNotifications();
-
-    const embedUrl = useMemo(() => {
-        if (window.location.origin.includes('localhost')) {
-            return getAppUrlFromApiUrl(API_URL, APPS.PROTONVERIFICATION);
-        }
-
-        return getAppUrlRelativeToOrigin(window.location.origin, APPS.PROTONVERIFICATION);
-    }, []);
-
-    useEffect(() => {
-        const handleMessage = (e: MessageEvent) => {
-            const { origin, data, source } = e;
-
-            const contentWindow = iframeRef.current?.contentWindow;
-
-            if (!contentWindow || origin !== embedUrl.origin || !data || source !== contentWindow) {
-                return;
-            }
-
-            const { type, payload } = JSON.parse(e.data);
-
-            switch (type) {
-                case 'RESIZE': {
-                    const { height } = payload;
-                    setIframeHeight(height);
-                    break;
-                }
-
-                case 'NOTIFICATION': {
-                    createNotification(payload);
-                    break;
-                }
-
-                case 'HUMAN_VERIFICATION_SUCCESS': {
-                    const { token, type } = payload;
-                    onSuccess(token, type);
-                    break;
-                }
-
-                default:
-            }
-        };
-
-        window.addEventListener('message', handleMessage);
-
-        return () => {
-            window.removeEventListener('message', handleMessage);
-        };
-    }, [onSuccess]);
-
-    const params = {
-        methods,
-        token,
-        defaultCountry: defaultCountry || undefined,
-        defaultEmail: defaultEmail || undefined,
-        defaultPhone: defaultPhone || undefined,
-        locale: localeCode,
-    };
-
-    const src = `${embedUrl.toString()}?${stringifySearchParams(params)}`;
-
-    return (
-        <iframe
-            style={{ height: `${iframeHeight}px`, width: '100%' }}
-            src={src}
-            ref={iframeRef}
-            title="verification-iframe"
-            sandbox="allow-scripts allow-same-origin allow-popups"
-        />
-    );
-};
-
-export default EmbeddedVerification;

diff --git a/packages/components/containers/verification/index.ts b/packages/components/containers/verification/index.ts
deleted file mode 100644
index 16a977e60..000000000
--- a/packages/components/containers/verification/index.ts
+++ /dev/null
@@ -1 +0,0 @@
-export { default as EmbeddedVerification } from './EmbeddedVerification';
