diff --git a/packages/components/hooks/useLinkHandler.tsx b/packages/components/hooks/useLinkHandler.tsx
index 30cda536c9..48d7c2c459 100644
--- a/packages/components/hooks/useLinkHandler.tsx
+++ b/packages/components/hooks/useLinkHandler.tsx
@@ -13,6 +13,7 @@ import isTruthy from '@proton/utils/isTruthy';
 import LinkConfirmationModal from '../components/notifications/LinkConfirmationModal/LinkConfirmationModal';
 import { getHostname, isExternal, punycodeUrl } from '../helpers/url';
 import { useHandler, useNotifications } from './index';
+import { openNewTab } from '@proton/shared/lib/helpers/browser';

 // Reference : Angular/src/app/utils/directives/linkHandler.js

@@ -130,6 +131,10 @@ export const useLinkHandler: UseLinkHandler = (
             return;
         }

+        event.preventDefault();
+        event.stopPropagation(); // Required for Safari
+        const link = punycodeUrl(src.encoded || src.raw);
+
         if (
             (askForConfirmation || isPhishingAttempt) &&
             isExternal(src.raw, window.location.hostname) &&
@@ -137,12 +142,10 @@ export const useLinkHandler: UseLinkHandler = (
                 .filter(isTruthy) // currentDomain can be null
                 .some((domain) => isSubDomain(hostname, domain))
         ) {
-            event.preventDefault();
-            event.stopPropagation(); // Required for Safari
-
-            const link = punycodeUrl(src.encoded || src.raw);
             setLink(link);
             setLinkConfirmationModalOpen(true);
+        } else {
+            openNewTab(link);
         }
     });

