diff --git a/packages/components/hooks/useLinkHandler.tsx b/packages/components/hooks/useLinkHandler.tsx
index 8fce50519e..f57dcb152b 100644
--- a/packages/components/hooks/useLinkHandler.tsx
+++ b/packages/components/hooks/useLinkHandler.tsx
@@ -14,6 +14,8 @@ import LinkConfirmationModal from '../components/notifications/LinkConfirmationM
 import { getHostname, isExternal, isSubDomain, punycodeUrl } from '../helpers/url';
 import { useHandler, useNotifications } from './index';

+import { openNewTab } from '@proton/shared/lib/helpers/browser';
+
 // Reference : Angular/src/app/utils/directives/linkHandler.js

 interface UseLinkHandlerOptions {
@@ -130,6 +132,10 @@ export const useLinkHandler: UseLinkHandler = (
             return;
         }

+        event.preventDefault();
+        event.stopPropagation(); // Required for Safari
+        const link = punycodeUrl(src.encoded || src.raw);
+
         if (
             (askForConfirmation || isPhishingAttempt) &&
             isExternal(src.raw, window.location.hostname) &&
@@ -137,12 +143,10 @@ export const useLinkHandler: UseLinkHandler = (
                 .filter(isTruthy) // currentDomain can be null
                 .some((domain) => isSubDomain(hostname, domain))
         ) {
-            event.preventDefault();
-            event.stopPropagation(); // Required for Safari
-
-            const link = punycodeUrl(src.encoded || src.raw);
             setLink(link);
             setLinkConfirmationModalOpen(true);
+        } else {
+            openNewTab(link);
         }
     });

