diff --git a/packages/components/hooks/useLinkHandler.tsx b/packages/components/hooks/useLinkHandler.tsx
index 854f8e4cde..e7bbbfc1c3 100644
--- a/packages/components/hooks/useLinkHandler.tsx
+++ b/packages/components/hooks/useLinkHandler.tsx
@@ -14,6 +14,7 @@ import isTruthy from '@proton/utils/isTruthy';
 import LinkConfirmationModal from '../components/notifications/LinkConfirmationModal/LinkConfirmationModal';
 import { getHostname, isExternal, punycodeUrl } from '../helpers/url';
 import useHandler from './useHandler';
+import { openNewTab } from '@proton/shared/lib/helpers/browser';

 // Reference : Angular/src/app/utils/directives/linkHandler.js

@@ -105,6 +106,10 @@ export const useLinkHandler: UseLinkHandler = (
             return;
         }

+        event.preventDefault();
+        event.stopPropagation(); // Required for Safari
+        const link = punycodeUrl(src.encoded || src.raw);
+
         if (
             (askForConfirmation || isPhishingAttempt) &&
             isExternal(src.raw, window.location.hostname) &&
@@ -112,12 +117,10 @@ export const useLinkHandler: UseLinkHandler = (
                 .filter(isTruthy) // currentDomain can be null
                 .some((domain) => isSubDomain(hostname, domain))
         ) {
-            event.preventDefault();
-            event.stopPropagation(); // Required for Safari
-
-            const link = punycodeUrl(src.encoded || src.raw);
             setLink(link);
             setLinkConfirmationModalOpen(true);
+        } else {
+            openNewTab(link);
         }
     });

