diff --git a/packages/pack/webpack.config.ts b/packages/pack/webpack.config.ts
index 63246f6565..ad7f8f4c1b 100644
--- a/packages/pack/webpack.config.ts
+++ b/packages/pack/webpack.config.ts
@@ -5,6 +5,7 @@ import 'webpack-dev-server';
 import { parseResource } from 'webpack/lib/util/identifier';

 import { getEntries } from './webpack/entries';
+import fs from 'fs';

 const getCssLoaders = require('./webpack/css.loader');
 const getAssetsLoaders = require('./webpack/assets.loader');
@@ -21,9 +22,7 @@ const getConfig = (env: any): Configuration => {

     const { getJsLoaders } = require(env.webpackOnCaffeine ? './webpack/js.loader.swc' : './webpack/js.loader');

-    const defaultBrowsersList = isProduction
-        ? `> 0.5%, not IE 11, Firefox ESR, Safari 14, iOS 14, Chrome 80`
-        : 'last 1 chrome version, last 1 firefox version, last 1 safari version';
+    const defaultBrowsersList = 'Chrome >= 126';

     const options = {
         isProduction,
@@ -33,8 +32,8 @@ const getConfig = (env: any): Configuration => {
         appMode: env.appMode || 'standalone',
         webpackOnCaffeine: env.webpackOnCaffeine,
         featureFlags: env.featureFlags || '',
-        writeSRI: env.writeSri !== 'false',
-        browserslist: env.browserslist ?? defaultBrowsersList,
+        writeSRI: false,
+        browserslist: defaultBrowsersList,
         buildData: {
             version: env.version,
             commit: env.commit,
@@ -53,8 +52,21 @@ const getConfig = (env: any): Configuration => {

     const version = options.buildData.version;

-    return {
-        target: `browserslist:${options.browserslist}`,
+    return ((): (value: Configuration) => Configuration => {
+        const file = path.resolve("./proton.config.js");
+        if (fs.existsSync(file)) {
+            console.log(
+                /*reset:*/"\x1b[0m" +
+                /*yellow:*/"\x1b[33m" +
+                ">>>" +
+                /*reset:*/"\x1b[0m",
+                `Found ${file}, extend the config`,
+            )
+            return eval("require")(file);
+        }
+        return (value: Configuration) => value;
+    })()({
+        target: [`browserslist:${defaultBrowsersList}`, "es5"],
         mode: isProduction ? 'production' : 'development',
         bail: isProduction,
         devtool: isProduction ? 'source-map' : 'cheap-module-source-map',
@@ -163,7 +175,7 @@ const getConfig = (env: any): Configuration => {
                 ],
             }),
         },
-    };
+    });
 };

 export default getConfig;

diff --git a/packages/shared/lib/webpack.constants.ts b/packages/shared/lib/webpack.constants.ts
index 9a14e94e98..ca7230f42a 100644
--- a/packages/shared/lib/webpack.constants.ts
+++ b/packages/shared/lib/webpack.constants.ts
@@ -3,8 +3,7 @@ declare const WEBPACK_FEATURE_FLAGS: string;
 export const FEATURE_FLAGS = typeof WEBPACK_FEATURE_FLAGS === 'undefined' ? '' : WEBPACK_FEATURE_FLAGS;

 declare const WEBPACK_APP_MODE: string;
-export const APP_MODE = typeof WEBPACK_APP_MODE === 'undefined' ? '' : WEBPACK_APP_MODE;
-export const appMode = APP_MODE === 'sso' ? 'sso' : 'standalone';
+export const appMode = typeof WEBPACK_APP_MODE === 'undefined' ? 'standalone' : WEBPACK_APP_MODE;

 declare const WEBPACK_PUBLIC_PATH: string;
 export const PUBLIC_PATH = typeof WEBPACK_PUBLIC_PATH === 'undefined' ? '' : WEBPACK_PUBLIC_PATH;

