diff --git a/packages/pack/lib/config.js b/packages/pack/lib/config.js
index b6484f719..a8ebc7fc0 100644
--- a/packages/pack/lib/config.js
+++ b/packages/pack/lib/config.js
@@ -129,7 +129,7 @@ const getConfigFile = ({ buildData, appData }) => {
     export const BRANCH = '${buildData.branch}';
     export const DATE_VERSION = '${buildData.date}';
     export const APP_NAME = '${appData.appName}';
-    export const API_URL =  '${(!appData.apiProxy && appData.api) || '/api'}';
+    export const API_URL = '${appData.api}';
     export const LOCALES = ${JSON.stringify(LOCALES)};
     export const VERSION_PATH = '${appData.publicPath}assets/version.json';
     export const SENTRY_DSN = '${appData.sentryDsn}';
