diff --git a/packages/pack/bin/protonPack.js b/packages/pack/bin/protonPack.js
index 86d8314c1..9dcb04131 100755
--- a/packages/pack/bin/protonPack.js
+++ b/packages/pack/bin/protonPack.js
@@ -68,6 +68,14 @@ const commandWithLog = (...args) => {

 addGlobalOptions(program.command('build').description('create an optimized production build'))
     .option('--no-sri', 'disable sri')
+    .option(
+        '--api <api>',
+        '',
+        (api) => {
+            return getApi(api);
+        },
+        getApi('')
+    )
     .action(async (options, env) => {
         console.log(chalk.magenta('Creating a production build...\n'));

diff --git a/packages/pack/lib/config.js b/packages/pack/lib/config.js
index 1b026386c..3d4546df2 100644
--- a/packages/pack/lib/config.js
+++ b/packages/pack/lib/config.js
@@ -128,7 +128,7 @@ const getConfigFile = ({ buildData, appData }) => {
     export const BRANCH = '${buildData.branch}';
     export const DATE_VERSION = '${buildData.date}';
     export const APP_NAME = '${appData.appName}';
-    export const API_URL = '/api';
+    export const API_URL = '${appData.api}';
     export const LOCALES = ${JSON.stringify(LOCALES)};
     export const VERSION_PATH = '${appData.publicPath}assets/version.json';
     export const SENTRY_DSN = '${appData.sentryDsn}';
