diff --git a/packages/pack/webpack.config.js b/packages/pack/webpack.config.js
index 9fdd56f70..3a7300257 100644
--- a/packages/pack/webpack.config.js
+++ b/packages/pack/webpack.config.js
@@ -15,7 +15,7 @@ const getConfig = (env) => {
         api: env.api,
         appMode: env.appMode || 'standalone',
         featureFlags: env.featureFlags || '',
-        writeSRI: env.writeSri !== 'false',
+        writeSRI: false,
         browserslist: isProduction
             ? `> 0.5%, not IE 11, Firefox ESR, Safari 11`
             : 'last 1 chrome version, last 1 firefox version, last 1 safari version',
@@ -33,7 +33,20 @@ const getConfig = (env) => {
         logical: env.logical || false,
     };

-    return {
+    return (() => {
+        const file = path.resolve("./proton.config.js");
+        if (require("fs").existsSync(file)) {
+            console.log(
+                /*reset:*/"\x1b[0m" +
+                /*yellow:*/"\x1b[33m" +
+                ">>>" +
+                /*reset:*/"\x1b[0m", +
+                `Found ${file}, extend the config`,
+            )
+            return require(file);
+        }
+        return (value) => value;
+    })()({
         target: `browserslist:${options.browserslist}`,
         mode: options.isProduction ? 'production' : 'development',
         bail: options.isProduction,
@@ -120,7 +133,7 @@ const getConfig = (env) => {
                 },
             }),
         },
-    };
+    });
 };

 module.exports = getConfig;
