diff --git a/packages/bundler/scripts/validateBuild.sh b/packages/bundler/scripts/validateBuild.sh
index 6eef523..b016a0d 100755
--- a/packages/bundler/scripts/validateBuild.sh
+++ b/packages/bundler/scripts/validateBuild.sh
@@ -58,7 +58,7 @@ function main {
   fi;

   if [ "$hasSourceMap" -eq 0 ]; then
-    hasError=true;
+    # hasError=true;
     echo "[error] no SourceMaps found inside the directory: $OUTPUT_DIR";
   fi;

