diff --git a/applications/account/package.json b/applications/account/package.json
index 3c5c6daff9..25828bb6b0 100644
--- a/applications/account/package.json
+++ b/applications/account/package.json
@@ -6,7 +6,6 @@
     "author": "",
     "main": "index.js",
     "scripts": {
-        "build": "cross-env NODE_ENV=production TS_NODE_PROJECT=\"../../tsconfig.webpack.json\" proton-pack build --appMode=sso",
         "check-types": "tsc",
         "i18n:upgrade": "proton-i18n extract --verbose && proton-i18n crowdin --verbose",
         "i18n:validate": "proton-i18n validate lint-functions",

diff --git a/applications/account/src/app/Setup.tsx b/applications/account/src/app/Setup.tsx
index 8f5e56ef0..a40d9dc17 100644
--- a/applications/account/src/app/Setup.tsx
+++ b/applications/account/src/app/Setup.tsx
@@ -2,14 +2,30 @@ import { PrivateAuthenticationStore, PublicAuthenticationStore, useAuthenticatio
 import { locales } from '@proton/shared/lib/i18n/locales';

 import PrivateApp from './content/PrivateApp';
-import PublicApp from './content/PublicApp';
+
+import * as _ElectronMail_React_import_ from 'react';
+
+const RedirectToOrigin = () => {
+    const timeoutMs = 5000;
+    const { origin: destination } = window.location;
+    const message = `Redirecting to "${destination}" in ${timeoutMs}ms`;
+    let timeoutId: undefined | ReturnType<typeof setTimeout>;
+    _ElectronMail_React_import_.useEffect(() => {
+        timeoutId = setTimeout(
+            () => window.location.replace(destination),
+            timeoutMs
+        );
+        return () => clearTimeout(timeoutId);
+    });
+    return _ElectronMail_React_import_.createElement(_ElectronMail_React_import_.Fragment, null, message);
+};

 const Setup = () => {
-    const { UID, login, logout } = useAuthentication() as PublicAuthenticationStore & PrivateAuthenticationStore;
+    const { UID, logout } = useAuthentication() as PublicAuthenticationStore & PrivateAuthenticationStore;
     if (UID) {
         return <PrivateApp locales={locales} onLogout={logout} />;
     }
-    return <PublicApp locales={locales} onLogin={login} />;
+    return <RedirectToOrigin />;
 };

 export default Setup;

diff --git a/applications/account/src/app/content/MainContainer.tsx b/applications/account/src/app/content/MainContainer.tsx
index 575b59dcec..5e774e01a1 100644
--- a/applications/account/src/app/content/MainContainer.tsx
+++ b/applications/account/src/app/content/MainContainer.tsx
@@ -132,7 +132,7 @@ const MainContainer = () => {
     const loadingFeatures = featuresFlags.some(({ loading }) => loading) || loadingDataRecovery;
     const recoveryNotification = useRecoveryNotification(false);

-    const appFromPathname = getAppFromPathnameSafe(location.pathname);
+    const appFromPathname = getAppFromPathnameSafe(window.location.pathname);
     const app = appFromPathname || getToApp(undefined, user);
     const appSlug = getSlugFromApp(app);

