diff --git a/node_modules/proton-bundler/scripts/validateBuild.sh b/node_modules/proton-bundler/scripts/validateBuild.sh
index 6eef523..b016a0d 100755
--- a/node_modules/proton-bundler/scripts/validateBuild.sh
+++ b/node_modules/proton-bundler/scripts/validateBuild.sh
@@ -58,7 +58,7 @@ function main {
   fi;

   if [ "$hasSourceMap" -eq 0 ]; then
-    hasError=true;
+    # hasError=true;
     echo "[error] no SourceMaps found inside the directory: $OUTPUT_DIR";
   fi;

diff --git a/node_modules/proton-shared/lib/apps/helper.ts b/node_modules/proton-shared/lib/apps/helper.ts
index b6913e1..31a52f7 100644
--- a/node_modules/proton-shared/lib/apps/helper.ts
+++ b/node_modules/proton-shared/lib/apps/helper.ts
@@ -32,7 +32,7 @@ export const getAppHrefBundle = (to: string, toApp: APP_NAMES) => {
     return `/${path}`;
 };

-export const getAccountSettingsApp = () => (isSSOMode ? APPS.PROTONACCOUNT : APPS.PROTONMAIL_SETTINGS);
+export const getAccountSettingsApp = () => APPS.PROTONACCOUNT;

 export const getClientID = (appName: APP_NAMES) => {
     return APPS_CONFIGURATION[appName].clientID;
diff --git a/node_modules/proton-shared/lib/constants.ts b/node_modules/proton-shared/lib/constants.ts
index ffc1441..9ffb7bd 100644
--- a/node_modules/proton-shared/lib/constants.ts
+++ b/node_modules/proton-shared/lib/constants.ts
@@ -33,22 +33,22 @@ export const APPS = {
 } as const;
 export const APPS_CONFIGURATION = {
     [APPS.PROTONACCOUNT]: {
-        publicPath: '',
-        subdomain: 'account',
+        publicPath: '/account',
+        subdomain: '',
         name: 'Proton Account',
         clientID: 'WebAccount',
         icon: 'protonaccount',
     },
     [APPS.PROTONMAIL]: {
         publicPath: '',
-        subdomain: 'beta',
+        subdomain: '',
         name: 'ProtonMail',
         clientID: 'WebMail',
         icon: 'protonmail',
     },
     [APPS.PROTONMAIL_SETTINGS]: {
         publicPath: '/settings',
-        subdomain: 'beta',
+        subdomain: '',
         name: 'ProtonMail',
         // Needs to be the same ClientID as mail since they are deployed as an embedded app
         clientID: 'WebMail',
@@ -56,21 +56,21 @@ export const APPS_CONFIGURATION = {
     },
     [APPS.PROTONCONTACTS]: {
         publicPath: '/contacts',
-        subdomain: 'contacts',
+        subdomain: '',
         name: 'ProtonContacts',
         clientID: 'WebContacts',
         icon: 'protoncontacts',
     },
     [APPS.PROTONDRIVE]: {
         publicPath: '/drive',
-        subdomain: 'drive',
+        subdomain: '',
         name: 'ProtonDrive',
         clientID: 'WebDrive',
         icon: 'protondrive',
     },
     [APPS.PROTONCALENDAR]: {
         publicPath: '/calendar',
-        subdomain: 'calendar',
+        subdomain: '',
         name: 'ProtonCalendar',
         clientID: 'WebCalendar',
         icon: 'protoncalendar',
