diff --git a/src/app/components/SharingModal/GeneratedLinkState.tsx b/src/app/components/SharingModal/GeneratedLinkState.tsx
index cdeec2f..89ff444 100644
--- a/src/app/components/SharingModal/GeneratedLinkState.tsx
+++ b/src/app/components/SharingModal/GeneratedLinkState.tsx
@@ -61,7 +61,7 @@ function GeneratedLinkState({
     const contentRef = useRef<HTMLDivElement>(null);
     const { createNotification } = useNotifications();
     const { openConfirmModal } = useConfirm();
-    const baseUrl = `${window.location.origin}/urls`;
+    const baseUrl = "https://drive.protonmail.com/urls";

     const [password, setPassword] = useState(initialPassword);
     const [expiration, setExpiration] = useState(initialExpiration);
