diff --git a/node_modules/react-components/containers/login/MinimalLoginContainer.tsx b/node_modules/react-components/containers/login/MinimalLoginContainer.tsx
index 3d56292..83a637c 100644
--- a/node_modules/react-components/containers/login/MinimalLoginContainer.tsx
+++ b/node_modules/react-components/containers/login/MinimalLoginContainer.tsx
@@ -14,6 +14,8 @@ import LoginTotpInput from './LoginTotpInput';
 import LoginUnlockInput from './LoginUnlockInput';
 import { FORM } from './interface';

+const ELECTRON_MAIL_FORM_STYLE = { padding: '0.8rem' } as const;
+
 interface Props extends Omit<UseLoginProps, 'api'> {
     needHelp?: React.ReactNode;
 }
@@ -51,7 +53,7 @@ const MinimalLoginContainer = ({ onLogin, ignoreUnlock = false, needHelp }: Prop
             withLoading(handleLogin().catch(catchHandler));
         };
         return (
-            <form name="loginForm" onSubmit={handleSubmit}>
+            <form name="loginForm" onSubmit={handleSubmit} style={ELECTRON_MAIL_FORM_STYLE}>
                 <Label htmlFor="login">{c('Label').t`Username or ProtonMail address`}</Label>
                 <div className="mb1">
                     <LoginUsernameInput
@@ -92,7 +94,7 @@ const MinimalLoginContainer = ({ onLogin, ignoreUnlock = false, needHelp }: Prop
             withLoading(handleTotp().catch(catchHandler));
         };
         return (
-            <form name="totpForm" onSubmit={handleSubmit}>
+            <form name="totpForm" onSubmit={handleSubmit} style={ELECTRON_MAIL_FORM_STYLE}>
                 <Label htmlFor="twoFa">{c('Label').t`Two-factor code`}</Label>
                 <div className="mb1">
                     <LoginTotpInput totp={totp} setTotp={loading ? noop : setters.totp} id="twoFa" />
@@ -118,7 +120,7 @@ const MinimalLoginContainer = ({ onLogin, ignoreUnlock = false, needHelp }: Prop
             withLoading(handleUnlock().catch(catchHandler));
         };
         return (
-            <form name="unlockForm" onSubmit={handleSubmit}>
+            <form name="unlockForm" onSubmit={handleSubmit} style={ELECTRON_MAIL_FORM_STYLE}>
                 <Label htmlFor="password">{c('Label').t`Mailbox password`}</Label>
                 <div className="mb1">
                     <LoginUnlockInput
