diff --git a/applications/drive/src/app/components/ShareLinkModal/GeneratedLinkState.tsx b/applications/drive/src/app/components/ShareLinkModal/GeneratedLinkState.tsx
index 3cb6545..4ad120b 100644
--- a/applications/drive/src/app/components/ShareLinkModal/GeneratedLinkState.tsx
+++ b/applications/drive/src/app/components/ShareLinkModal/GeneratedLinkState.tsx
@@ -73,7 +73,7 @@ function GeneratedLinkState({
 }: Props) {
     const contentRef = useRef<HTMLDivElement>(null);
     const { createNotification } = useNotifications();
-    const baseUrl = `${window.location.origin}/urls`;
+    const baseUrl = "https://drive.protonmail.com/urls";

     const [password, setPassword] = useState(customPassword);
     const [expiration, setExpiration] = useState(initialExpiration);
