diff --git a/src/app/components/SharingModal/GeneratedLinkState.tsx b/src/app/components/SharingModal/GeneratedLinkState.tsx
index c5f63bb..b477290 100644
--- a/src/app/components/SharingModal/GeneratedLinkState.tsx
+++ b/src/app/components/SharingModal/GeneratedLinkState.tsx
@@ -60,7 +60,7 @@ function GeneratedLinkState({
     const contentRef = useRef<HTMLDivElement>(null);
     const { createNotification } = useNotifications();
     const { openConfirmModal } = useConfirm();
-    const baseUrl = `${window.location.origin}/urls`;
+    const baseUrl = "https://drive.protonmail.com/urls";

     const [password, setPassword] = useState(customPassword);
     const [expiration, setExpiration] = useState(initialExpiration);
