diff --git a/applications/drive/src/app/components/SharingModal/GeneratedLinkState.tsx b/applications/drive/src/app/components/SharingModal/GeneratedLinkState.tsx
index 9308bf1..ff1b976 100644
--- a/applications/drive/src/app/components/SharingModal/GeneratedLinkState.tsx
+++ b/applications/drive/src/app/components/SharingModal/GeneratedLinkState.tsx
@@ -72,7 +72,7 @@ function GeneratedLinkState({
 }: Props) {
     const contentRef = useRef<HTMLDivElement>(null);
     const { createNotification } = useNotifications();
-    const baseUrl = `${window.location.origin}/urls`;
+    const baseUrl = "https://drive.protonmail.com/urls";

     const [password, setPassword] = useState(customPassword);
     const [expiration, setExpiration] = useState(initialExpiration);
