diff --git a/src/app/Setup.tsx b/src/app/Setup.tsx
index 9d333b8..2921f49 100644
--- a/src/app/Setup.tsx
+++ b/src/app/Setup.tsx
@@ -4,14 +4,29 @@ import locales from 'proton-shared/lib/i18n/locales';
 import { PublicAuthenticationStore, PrivateAuthenticationStore, useAuthentication } from 'react-components';

 import PrivateApp from './content/PrivateApp';
-import PublicApp from './content/PublicApp';
+
+const RedirectToOrigin = () => {
+    const timeoutMs = 5000;
+    const {origin: destination} = window.location;
+    const message = `Redirecting to "${destination}" in ${timeoutMs}ms`;
+    let timeoutId: undefined | ReturnType<typeof setTimeout>;
+    React.useEffect(() => {
+        timeoutId = setTimeout(
+            () => window.location.replace(destination),
+            timeoutMs,
+        );
+        return () => clearTimeout(timeoutId);
+    });
+    return React.createElement(React.Fragment, null, message);
+}

 const Setup = () => {
-    const { UID, login, logout } = useAuthentication() as PublicAuthenticationStore & PrivateAuthenticationStore;
+    const { UID, logout } = useAuthentication() as PublicAuthenticationStore & PrivateAuthenticationStore;
     if (UID) {
         return <PrivateApp locales={locales} onLogout={logout} />;
     }
-    return <PublicApp locales={locales} onLogin={login} />;
+
+    return <RedirectToOrigin/>;
 };

 export default Setup;
diff --git a/node_modules/proton-shared/lib/helpers/sentry.js b/node_modules/proton-shared/lib/helpers/sentry.js
index 56310c9..50b19e1 100644
--- a/node_modules/proton-shared/lib/helpers/sentry.js
+++ b/node_modules/proton-shared/lib/helpers/sentry.js
@@ -1,32 +1,7 @@
-import * as Sentry from '@sentry/browser';
-
-const isLocalhost = () => document.location.host.startsWith('localhost');
-
-const isProduction = (host) => host === 'mail.protonmail.com' || host === 'account.protonvpn.com';
-
 function main({ SENTRY_DSN, COMMIT_RELEASE, APP_VERSION }) {
-    // No need to configure it if we don't load the DSN
-    if (!SENTRY_DSN || isLocalhost()) {
-        return;
-    }
-
-    const { host } = document.location;
-
-    // Assumes SENTRY_DSN is: https://111b3eeaaec34cae8e812df705690a36@sentry/11
-    // To get https://111b3eeaaec34cae8e812df705690a36@mail.protonmail.com/api/reports/sentry/11
-    const dsn = SENTRY_DSN.replace('sentry', `${host}/api/reports/sentry`);
-
-    Sentry.init({
-        dsn,
-        release: isProduction(host) ? APP_VERSION : COMMIT_RELEASE,
-        environment: host,
-    });

-    Sentry.configureScope((scope) => {
-        scope.setTag('appVersion', APP_VERSION);
-    });
 }

-export const traceError = (e) => !isLocalhost() && Sentry.captureException(e);
+export const traceError = (e) => console.error(e);

 export default main;
