diff --git a/packages/shared/lib/helpers/sentry.ts b/packages/shared/lib/helpers/sentry.ts
index 8b6a27111d..96055ca4f6 100644
--- a/packages/shared/lib/helpers/sentry.ts
+++ b/packages/shared/lib/helpers/sentry.ts
@@ -1,27 +1,17 @@
 import type { BrowserOptions, SeverityLevel, Span } from '@sentry/browser';
 import {
-    Integrations as SentryIntegrations,
-    addBreadcrumb,
     captureException,
     configureScope,
-    init,
-    makeFetchTransport,
     captureMessage as sentryCaptureMessage,
-    startInactiveSpan as sentryStartInactiveSpan,
 } from '@sentry/browser';
-import type { BrowserTransportOptions } from '@sentry/browser/types/transports/types';
 import type { Breadcrumb } from '@sentry/types/types/breadcrumb';
 import type { ErrorEvent, EventHint } from '@sentry/types/types/event';

-import { Availability, AvailabilityTypes } from '@proton/utils/availability';
-
 import { getApiError, getIsConnectionIssue } from '../api/helpers/apiErrorHelper';
 import { VPN_HOSTNAME } from '../constants';
-import { SHARE_GENERATED_PASSWORD_LENGTH } from '../drive/constants';
 import { ApiError } from '../fetch/ApiError';
 import { getUIDHeaders } from '../fetch/headers';
 import type { ProtonConfig } from '../interfaces';
-import { isElectronApp } from './desktop';

 export { spanToJSON } from '@sentry/core';
 export type { Span, SeverityLevel } from '@sentry/browser';
@@ -82,21 +72,6 @@ export const getContentTypeHeaders = (input: FirstFetchParameter): HeadersInit =
     return {};
 };

-const sentryFetch: typeof fetch = (input, init?) => {
-    return globalThis.fetch(input, {
-        ...init,
-        headers: {
-            ...init?.headers,
-            ...getContentTypeHeaders(input),
-            ...context.authHeaders,
-        },
-    });
-};
-
-const makeProtonFetchTransport = (options: BrowserTransportOptions) => {
-    return makeFetchTransport(options, sentryFetch);
-};
-
 const isLocalhost = (host: string) => host.startsWith('localhost');
 export const isProduction = (host: string) =>
     host.endsWith('.proton.me') || host.endsWith('.protonvpn.com') || host === VPN_HOSTNAME;
@@ -194,133 +169,7 @@ function main({
     UID,
     beforeSendIgnore,
     setupIgnore = ({ host }) => isLocalhost(host),
-}: SentryOptions) {
-    const { SENTRY_DSN, SENTRY_DESKTOP_DSN, APP_VERSION } = config;
-    const sentryDSN = isElectronApp ? SENTRY_DESKTOP_DSN || SENTRY_DSN : SENTRY_DSN;
-    const { host, release, environment } = sentryConfig;
-
-    // No need to configure it if we don't load the DSN
-    if (!sentryDSN || setupIgnore(sentryConfig)) {
-        return;
-    }
-
-    setUID(UID);
-
-    // Assumes sentryDSN is: https://111b3eeaaec34cae8e812df705690a36@sentry/11
-    // To get https://111b3eeaaec34cae8e812df705690a36@protonmail.com/api/core/v4/reports/sentry/11
-    const dsn = sentryDSN.replace('sentry', `${host}/api/core/v4/reports/sentry`);
-
-    init({
-        dsn,
-        release,
-        environment,
-        normalizeDepth: 5,
-        transport: makeProtonFetchTransport,
-        autoSessionTracking: sessionTracking,
-        // do not log calls to console.log, console.error, etc.
-        integrations: [
-            new SentryIntegrations.Breadcrumbs({
-                console: false,
-            }),
-        ],
-        // Disable client reports. Client reports are used by sentry to retry events that failed to send on visibility change.
-        // Unfortunately Sentry does not use the custom transport for those, and thus fails to add the headers the API requires.
-        sendClientReports: false,
-        beforeSend(event, hint) {
-            if (beforeSendIgnore?.(event, hint)) {
-                return null;
-            }
-
-            const error = hint?.originalException as any;
-            const stack = typeof error === 'string' ? error : error?.stack;
-            // Filter out broken ferdi errors
-            if (stack && stack.match(/ferdi|franz/i)) {
-                return null;
-            }
-
-            // Not interested in uncaught API errors, or known errors
-            if (error instanceof ApiError || error?.trace === false) {
-                return null;
-            }
-
-            if (!context.enabled) {
-                return null;
-            }
-
-            // Remove the hash from the request URL and navigation breadcrumbs to avoid
-            // leaking the search parameters of encrypted searches
-            if (event.request && event.request.url) {
-                [event.request.url] = event.request.url.split('#');
-            }
-            // keys/all endpoint accepts Email as parameter which is PII.
-            if (event.request && event.request.url) {
-                [event.request.url] = event.request.url.toLowerCase().split('email');
-            }
-            if (event.breadcrumbs) {
-                event.breadcrumbs = event.breadcrumbs.map((breadcrumb) => {
-                    if (breadcrumb.category === 'navigation' && breadcrumb.data) {
-                        [breadcrumb.data.from] = breadcrumb.data.from.split('#');
-                        [breadcrumb.data.to] = breadcrumb.data.to.split('#');
-                    }
-
-                    if (breadcrumb.category === 'fetch' && breadcrumb.data) {
-                        // Redact email addresses in breadcrumb messages
-                        breadcrumb.message = breadcrumb.data.url.replace(/Email=[^&]+/, 'Email=[Filtered]');
-                    }
-
-                    // Button titles may contain accidental PII
-                    if (
-                        breadcrumb.category === 'ui.click' &&
-                        breadcrumb.message &&
-                        breadcrumb.message.startsWith('button')
-                    ) {
-                        breadcrumb.message = breadcrumb.message.replace(/\[title=".+?"\]/g, '[title="(Filtered)"]');
-                    }
-
-                    return breadcrumb;
-                });
-            }
-
-            // This will prevent from sending the password in the public url. Ex: https://drive.proton.dev/urls/TBYS1295RG#[password]
-            if (event.exception && event.exception.values) {
-                event.exception.values = event.exception.values.map((error) => {
-                    if (
-                        error.value &&
-                        error.value.includes('/urls/') &&
-                        error.value.match(new RegExp(`#[a-zA-Z0-9]{${SHARE_GENERATED_PASSWORD_LENGTH}}\\b`))
-                    ) {
-                        // Redact sensitive information in the error message
-                        error.value = error.value.replace(
-                            new RegExp(`#[a-zA-Z0-9]{${SHARE_GENERATED_PASSWORD_LENGTH}}\\b.*$`),
-                            '#[Filtered]'
-                        );
-                    }
-
-                    return error;
-                });
-            }
-
-            // This should help provide more context for Promise rejections that throw an Event object instead of an Error object.
-            if (hint && hint.originalException instanceof Event) {
-                event.extra = event.extra || {};
-                event.extra.originalEvent = {
-                    type: hint.originalException.type,
-                    target: String(hint.originalException.target),
-                };
-            }
-
-            return event;
-        },
-        // Some ignoreErrors and denyUrls are taken from this gist: https://gist.github.com/Chocksy/e9b2cdd4afc2aadc7989762c4b8b495a
-        // This gist is suggested in the Sentry documentation: https://docs.sentry.io/clients/javascript/tips/#decluttering-sentry
-        ignoreErrors,
-        denyUrls,
-    });
-
-    configureScope((scope) => {
-        scope.setTag('appVersion', APP_VERSION);
-    });
-}
+}: SentryOptions) {}

 export const setSentryTags = (tags: Record<string, any>) => {
     configureScope((scope) => {
@@ -330,25 +179,11 @@ export const setSentryTags = (tags: Record<string, any>) => {
     });
 };

-export const traceError = (...args: Parameters<typeof captureException>) => {
-    if (!isLocalhost(self.location.host)) {
-        captureException(...args);
-        Availability.mark(AvailabilityTypes.SENTRY);
-    }
-};
+export const traceError = (...args: Parameters<typeof captureException>) => {};

-export const addSentryBreadcrumb = (breadcrumb: Breadcrumb): void => {
-    if (typeof self === 'undefined' || isLocalhost(self.location.host)) {
-        return;
-    }
-    addBreadcrumb(breadcrumb);
-};
+export const addSentryBreadcrumb = (breadcrumb: Breadcrumb): void => {};

-export const captureMessage = (...args: Parameters<typeof sentryCaptureMessage>) => {
-    if (!isLocalhost(self.location.host)) {
-        sentryCaptureMessage(...args);
-    }
-};
+export const captureMessage = (...args: Parameters<typeof sentryCaptureMessage>) => {};

 export const getSentryError = (error: any): any => {
     if (error instanceof ApiError) {
@@ -421,17 +256,7 @@ export const traceInitiativeError = (
     initiative: SentryInitiative,
     error: CaptureExceptionArgs[0],
     context?: InitiativeContext
-) => {
-    if (!isLocalhost(self.location.host)) {
-        captureException(error, {
-            ...context,
-            tags: {
-                ...context?.tags,
-                initiative,
-            },
-        });
-    }
-};
+) => console.error(initiative, error);

 /**
  * Capture message with an additional initiative tag
@@ -440,26 +265,12 @@ export const captureInitiativeMessage = (
     initiative: SentryInitiative,
     message: string,
     context?: InitiativeContext
-) => {
-    captureMessage(message, {
-        ...context,
-        tags: {
-            ...context?.tags,
-            initiative,
-        },
-    });
-};
+) => console.log(initiative, message);

 export const startInactiveSpan = (
     name: string,
     initiative: SentryInitiative,
     tags?: Record<string, string>
-): Span | undefined => {
-    return sentryStartInactiveSpan({
-        name,
-        op: initiative,
-        tags,
-    });
-};
+): Span | undefined => console.log(name, initiative, tags);

 export default main;
--
