diff --git a/packages/shared/lib/helpers/sentry.ts b/packages/shared/lib/helpers/sentry.ts
index 10fbb88611..4a7ba54f7a 100644
--- a/packages/shared/lib/helpers/sentry.ts
+++ b/packages/shared/lib/helpers/sentry.ts
@@ -181,108 +181,14 @@ function main({
     ignore = ({ host }) => isLocalhost(host),
     denyUrls = getDefaultDenyUrls(),
     ignoreErrors = getDefaultIgnoreErrors(),
-}: SentryOptions) {
-    const { SENTRY_DSN, SENTRY_DESKTOP_DSN, APP_VERSION } = config;
-    const sentryDSN = isElectronApp ? SENTRY_DESKTOP_DSN || SENTRY_DSN : SENTRY_DSN;
-    const { host, release, environment } = sentryConfig;
-
-    // No need to configure it if we don't load the DSN
-    if (!sentryDSN || ignore(sentryConfig)) {
-        return;
-    }
-
-    setUID(UID);
-
-    // Assumes sentryDSN is: https://111b3eeaaec34cae8e812df705690a36@sentry/11
-    // To get https://111b3eeaaec34cae8e812df705690a36@protonmail.com/api/core/v4/reports/sentry/11
-    const dsn = sentryDSN.replace('sentry', `${host}/api/core/v4/reports/sentry`);
-
-    init({
-        dsn,
-        release,
-        environment,
-        normalizeDepth: 5,
-        transport: makeProtonFetchTransport,
-        autoSessionTracking: sessionTracking,
-        // do not log calls to console.log, console.error, etc.
-        integrations: [
-            new SentryIntegrations.Breadcrumbs({
-                console: false,
-            }),
-        ],
-        // Disable client reports. Client reports are used by sentry to retry events that failed to send on visibility change.
-        // Unfortunately Sentry does not use the custom transport for those, and thus fails to add the headers the API requires.
-        sendClientReports: false,
-        beforeSend(event, hint) {
-            const error = hint?.originalException as any;
-            const stack = typeof error === 'string' ? error : error?.stack;
-            // Filter out broken ferdi errors
-            if (stack && stack.match(/ferdi|franz/i)) {
-                return null;
-            }
-
-            // Not interested in uncaught API errors, or known errors
-            if (error instanceof ApiError || error?.trace === false) {
-                return null;
-            }
-
-            if (!context.enabled) {
-                return null;
-            }
-
-            // Remove the hash from the request URL and navigation breadcrumbs to avoid
-            // leaking the search parameters of encrypted searches
-            if (event.request && event.request.url) {
-                [event.request.url] = event.request.url.split('#');
-            }
-            // keys/all endpoint accepts Email as parameter which is PII.
-            if (event.request && event.request.url) {
-                [event.request.url] = event.request.url.toLowerCase().split('email');
-            }
-            if (event.breadcrumbs) {
-                event.breadcrumbs = event.breadcrumbs.map((breadcrumb) => {
-                    if (breadcrumb.category === 'navigation' && breadcrumb.data) {
-                        [breadcrumb.data.from] = breadcrumb.data.from.split('#');
-                        [breadcrumb.data.to] = breadcrumb.data.to.split('#');
-                    }
-
-                    // Button titles may contain accidental PII
-                    if (
-                        breadcrumb.category === 'ui.click' &&
-                        breadcrumb.message &&
-                        breadcrumb.message.startsWith('button')
-                    ) {
-                        breadcrumb.message = breadcrumb.message.replace(/\[title=".+?"\]/g, '[title="(Filtered)"]');
-                    }
-
-                    return breadcrumb;
-                });
-            }
-
-            return event;
-        },
-        // Some ignoreErrors and denyUrls are taken from this gist: https://gist.github.com/Chocksy/e9b2cdd4afc2aadc7989762c4b8b495a
-        // This gist is suggested in the Sentry documentation: https://docs.sentry.io/clients/javascript/tips/#decluttering-sentry
-        ignoreErrors,
-        denyUrls,
-    });
-
-    configureScope((scope) => {
-        scope.setTag('appVersion', APP_VERSION);
-    });
-}
+}: SentryOptions) {}

 export const traceError = (...args: Parameters<typeof captureException>) => {
-    if (!isLocalhost(window.location.host)) {
-        captureException(...args);
-        Availability.mark(AvailabilityTypes.SENTRY);
-    }
+    console.error(...args);
 };

 export const captureMessage = (...args: Parameters<typeof sentryCaptureMessage>) => {
-    if (!isLocalhost(window.location.host)) {
-        sentryCaptureMessage(...args);
-    }
+    console.log(...args);
 };

 type MailInitiative = 'drawer-security-center' | 'composer' | 'assistant' | 'mail-onboarding';
@@ -295,13 +201,7 @@ type CaptureExceptionArgs = Parameters<typeof captureException>;
  * @param error
  */
 export const traceInitiativeError = (initiative: MailInitiative, error: CaptureExceptionArgs[0]) => {
-    if (!isLocalhost(window.location.host)) {
-        captureException(error, {
-            tags: {
-                initiative,
-            },
-        });
-    }
+    console.error(initiative, error);
 };

 /**
@@ -313,11 +213,7 @@ export const captureInitiativeMessage: (initiative: SentryInitiative, message: s
     initiative,
     message
 ) => {
-    captureMessage(message, {
-        tags: {
-            initiative,
-        },
-    });
+    console.log(initiative, message);
 };

 export default main;
