diff --git a/packages/shared/lib/helpers/url.ts b/packages/shared/lib/helpers/url.ts
index 0c21c0f042..8ccd5ce82b 100644
--- a/packages/shared/lib/helpers/url.ts
+++ b/packages/shared/lib/helpers/url.ts
@@ -183,48 +183,16 @@ export const getSecondLevelDomain = (hostname: string) => {
     return hostname.slice(hostname.indexOf('.') + 1);
 };

-export const getRelativeApiHostname = (hostname: string) => {
-    const idx = hostname.indexOf('.');
-    const first = hostname.slice(0, idx);
-    const second = hostname.slice(idx + 1);
-    return `${first}-api.${second}`;
-};
-
 export const getIsDohDomain = (origin: string) => {
     return DOH_DOMAINS.some((dohDomain) => origin.endsWith(dohDomain));
 };

 export const getApiSubdomainUrl = (pathname: string) => {
-    const url = new URL('/', window.location.origin);
-    if (url.hostname === 'localhost' || getIsDohDomain(url.origin)) {
-        url.pathname = `/api${pathname}`;
-        return url;
-    }
-    url.hostname = getRelativeApiHostname(url.hostname);
+    const url = new URL('/', '___ELECTRON_MAIL_PROTON_API_ENTRY_URL_PLACEHOLDER___');
     url.pathname = pathname;
     return url;
 };

-export const getAppUrlFromApiUrl = (apiUrl: string, appName: APP_NAMES) => {
-    const { subdomain } = APPS_CONFIGURATION[appName];
-    const url = new URL(apiUrl);
-    const { hostname } = url;
-    const index = hostname.indexOf('.');
-    const tail = hostname.slice(index + 1);
-    url.pathname = '';
-    url.hostname = `${subdomain}.${tail}`;
-    return url;
-};
-
-export const getAppUrlRelativeToOrigin = (origin: string, appName: APP_NAMES) => {
-    const { subdomain } = APPS_CONFIGURATION[appName];
-    const url = new URL(origin);
-    const segments = url.host.split('.');
-    segments[0] = subdomain;
-    url.hostname = segments.join('.');
-    return url;
-};
-
 let cache = '';
 export const getStaticURL = (path: string) => {
     if (

diff --git a/packages/shared/lib/fetch/helpers.ts b/packages/shared/lib/fetch/helpers.ts
index 666b9f56c6..246f0e03b4 100644
--- a/packages/shared/lib/fetch/helpers.ts
+++ b/packages/shared/lib/fetch/helpers.ts
@@ -11,6 +11,7 @@ const appendQueryParams = (url: URL, params: { [key: string]: any }) => {
     });
 };

+/* <electron-mail-mark> */
 export const createUrl = (urlString: string, params: { [key: string]: any } = {}) => {
     let url: URL;
     if (typeof window !== 'undefined') {
@@ -21,6 +22,7 @@ export const createUrl = (urlString: string, params: { [key: string]: any } = {}
     appendQueryParams(url, params);
     return url;
 };
+/* </electron-mail-mark> */

 export const checkStatus = (response: Response, config: any) => {
     const { status } = response;
