diff --git a/lib/src/clang-fetcher.js b/lib/src/clang-fetcher.js
index ed9243f66d49658e9a03654fa1042cfa1dd06658..98916a395db0a88a10e52e29cd58ce6baac8119f 100644
--- a/lib/src/clang-fetcher.js
+++ b/lib/src/clang-fetcher.js
@@ -81,7 +81,7 @@ async function getClangEnvironmentVars(electronVersion, targetArch) {
         env: {
             CC: `"${path.resolve(clangDir, 'clang')}" ${clangArgs.join(' ')}`,
             CXX: `"${path.resolve(clangDir, 'clang++')}" ${clangArgs.join(' ')}`,
-            CFLAGS: `${cxxflags.join(' ')}`,
+            CFLAGS: `${process.env.CFLAGS ? process.env.CFLAGS + " ": ""}${cxxflags.join(' ')}`,
             CXXFLAGS: `${cxxflags.join(' ')}`
         },
         args: gypArgs,
diff --git a/lib/src/module-rebuilder.js b/lib/src/module-rebuilder.js
index 0d78dbe1310ebecf16f96e24ab683e3d9e4912a6..49a4111aa26538bc95919e7486006d5c80be5f52 100644
--- a/lib/src/module-rebuilder.js
+++ b/lib/src/module-rebuilder.js
@@ -65,7 +65,8 @@ class ModuleRebuilder {
      * Whether a prebuild-install-generated native module exists.
      */
     async prebuildInstallNativeModuleExists() {
-        return this.prebuildInstall.prebuiltModuleExists();
+        // return this.prebuildInstall.prebuiltModuleExists();
+        return false;
     }
     /**
      * If the native module uses prebuildify, check to see if it comes with a prebuilt module for