diff --git a/lib/module-rebuilder.js b/lib/module-rebuilder.js
index c5ff1369350fac9048eeafda3fb309edfbfd4fbe..53ef3bb28dfd73a7320202838370dc35cb1a3459 100644
--- a/lib/module-rebuilder.js
+++ b/lib/module-rebuilder.js
@@ -45,7 +45,7 @@ export class ModuleRebuilder {
      * Whether a prebuild-install-generated native module exists.
      */
     async prebuildInstallNativeModuleExists() {
-        return this.prebuildInstall.prebuiltModuleExists();
+        return false;
     }
     /**
      * If the native module uses prebuildify, check to see if it comes with a prebuilt module for
