diff --git a/dist/emscripten-module.mjs b/dist/emscripten-module.mjs
index 99ce2e2ba92cc7c432bbdf02ef072f4bc9a493a6..5ef3d1c61287c09ae48ab0a87e39c9a73dd144de 100644
--- a/dist/emscripten-module.mjs
+++ b/dist/emscripten-module.mjs
@@ -1,12 +1,12 @@
 
 var QuickJSRaw = (() => {
-  var _scriptName = import.meta.url;
-  
+  var _scriptName;
+
   return (
 async function(moduleArg = {}) {
   var moduleRtn;
 
-var c=moduleArg,aa,f,ba=new Promise((a,b)=>{aa=a;f=b}),ca="object"==typeof window,n="function"==typeof importScripts,p="object"==typeof process&&"object"==typeof process.versions&&"string"==typeof process.versions.node;if(p){const {createRequire:a}=await import("module");var require=a(import.meta.url)}function r(a){a={log:a||function(){}};for(const b of r.Ia)b(a);return c.quickJSEmscriptenExtensions=a}r.Ia=[];c.quickjsEmscriptenInit=r;r.Ia.push(a=>{a.getWasmMemory=function(){return t}});
+var c=moduleArg,aa,f,ba=new Promise((a,b)=>{aa=a;f=b}),ca="object"==typeof window,n="function"==typeof importScripts,p="object"==typeof process&&"object"==typeof process.versions&&"string"==typeof process.versions.node;function r(a){a={log:a||function(){}};for(const b of r.Ia)b(a);return c.quickJSEmscriptenExtensions=a}r.Ia=[];c.quickjsEmscriptenInit=r;r.Ia.push(a=>{a.getWasmMemory=function(){return t}});
 var da=Object.assign({},c),u="./this.program",w=(a,b)=>{throw b;},x="",y,z;
 if(p){var fs=require("fs"),ea=require("path");x=require("url").fileURLToPath(new URL("./",import.meta.url));z=a=>{a=A(a)?new URL(a):ea.normalize(a);return fs.readFileSync(a)};y=a=>{a=A(a)?new URL(a):ea.normalize(a);return new Promise((b,d)=>{fs.readFile(a,void 0,(e,g)=>{e?d(e):b(g.buffer)})})};!c.thisProgram&&1<process.argv.length&&(u=process.argv[1].replace(/\\/g,"/"));process.argv.slice(2);w=(a,b)=>{process.exitCode=a;throw b;}}else if(ca||n)n?x=self.location.href:"undefined"!=typeof document&&
 document.currentScript&&(x=document.currentScript.src),_scriptName&&(x=_scriptName),x.startsWith("blob:")?x="":x=x.substr(0,x.replace(/[?#].*/,"").lastIndexOf("/")+1),n&&(z=a=>{var b=new XMLHttpRequest;b.open("GET",a,!1);b.responseType="arraybuffer";b.send(null);return new Uint8Array(b.response)}),y=a=>A(a)?new Promise((b,d)=>{var e=new XMLHttpRequest;e.open("GET",a,!0);e.responseType="arraybuffer";e.onload=()=>{200==e.status||0==e.status&&e.response?b(e.response):d(e.status)};e.onerror=d;e.send(null)}):
