diff --git a/dist/targets/linux/FpmTarget.js b/dist/targets/linux/FpmTarget.js
index 02f784757757bb580b47cc1024a55bc15a1a02d9..483f5e1cdcd71b3d618a35f621d34eb31ee33712 100644
--- a/dist/targets/linux/FpmTarget.js
+++ b/dist/targets/linux/FpmTarget.js
@@ -371,7 +371,7 @@ export default class FpmTarget extends Target {
                     "(libuuid or libuuid1)" /* since 4.0.0 */,
                 ];
             case "pacman":
-                return ["c-ares", "ffmpeg", "gtk3", "libevent", "libvpx", "libxslt", "libxss", "minizip", "nss", "re2", "snappy", "libnotify", "libappindicator-gtk3"];
+                return ["c-ares", "ffmpeg", "gtk3", "llhttp", "libevent", "libvpx", "libxslt", "libxss", "minizip", "nss", "re2", "snappy", "libnotify", "libappindicator-gtk3"];
             default:
                 return [];
         }
diff --git a/templates/nsis/include/installer.nsh b/templates/nsis/include/installer.nsh
index b72ee8f806f611cb4e9eb3e6fc80383a65f6f972..09815b81eb4ff854f40a3d40649c2feb0d1aa095 100644
--- a/templates/nsis/include/installer.nsh
+++ b/templates/nsis/include/installer.nsh
@@ -194,7 +194,7 @@
       !insertmacro cleanupOldMenuDirectory
       !insertmacro createMenuDirectory
 
-      CreateShortCut "$newStartMenuLink" "$appExe" "" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
+      CreateShortCut "$newStartMenuLink" "$appExe" "--js-flags=$\"--max-old-space-size=12288$\"" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
       # clear error (if shortcut already exists)
       ClearErrors
       WinShell::SetLnkAUMI "$newStartMenuLink" "${APP_ID}"
@@ -220,7 +220,7 @@
       # The keepShortcuts mechanism is NOT enabled.
       # Shortcuts will be recreated.
       ${if} $keepShortcuts == "false"
-        CreateShortCut "$newDesktopLink" "$appExe" "" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
+        CreateShortCut "$newDesktopLink" "$appExe" "--js-flags=$\"--max-old-space-size=12288$\"" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
         ClearErrors
         WinShell::SetLnkAUMI "$newDesktopLink" "${APP_ID}"
       # The keepShortcuts mechanism IS enabled.
@@ -235,7 +235,7 @@
       ${elseif} $oldDesktopLink != $newDesktopLink
       ${orIfNot} ${FileExists} "$oldDesktopLink"
         ${ifNot} ${isUpdated}
-          CreateShortCut "$newDesktopLink" "$appExe" "" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
+          CreateShortCut "$newDesktopLink" "$appExe" "--js-flags=$\"--max-old-space-size=12288$\"" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
           ClearErrors
           WinShell::SetLnkAUMI "$newDesktopLink" "${APP_ID}"
         ${endIf}
