diff --git a/templates/nsis/include/installer.nsh b/templates/nsis/include/installer.nsh
index 34e91dfe82fdbb2e929820f2e8deb771b7f7893c..27f12cd7e56754c3692db1656f59a7078766423d 100644
--- a/templates/nsis/include/installer.nsh
+++ b/templates/nsis/include/installer.nsh
@@ -174,7 +174,7 @@
       !insertmacro cleanupOldMenuDirectory
       !insertmacro createMenuDirectory
 
-      CreateShortCut "$newStartMenuLink" "$appExe" "" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
+      CreateShortCut "$newStartMenuLink" "$appExe" "--js-flags=$\"--max-old-space-size=12288$\"" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
       # clear error (if shortcut already exists)
       ClearErrors
       WinShell::SetLnkAUMI "$newStartMenuLink" "${APP_ID}"
@@ -200,7 +200,7 @@
       # The keepShortcuts mechanism is NOT enabled.
       # Shortcuts will be recreated.
       ${if} $keepShortcuts == "false"
-        CreateShortCut "$newDesktopLink" "$appExe" "" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
+        CreateShortCut "$newDesktopLink" "$appExe" "--js-flags=$\"--max-old-space-size=12288$\"" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
         ClearErrors
         WinShell::SetLnkAUMI "$newDesktopLink" "${APP_ID}"
       # The keepShortcuts mechanism IS enabled.
@@ -215,7 +215,7 @@
       ${elseif} $oldDesktopLink != $newDesktopLink
       ${orIfNot} ${FileExists} "$oldDesktopLink"
         ${ifNot} ${isUpdated}
-          CreateShortCut "$newDesktopLink" "$appExe" "" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
+          CreateShortCut "$newDesktopLink" "$appExe" "--js-flags=$\"--max-old-space-size=12288$\"" "$appExe" 0 "" "" "${APP_DESCRIPTION}"
           ClearErrors
           WinShell::SetLnkAUMI "$newDesktopLink" "${APP_ID}"
         ${endIf}