diff --git a/index.js b/index.js
index 970dfa2226a9ece95581d6fd8a93cf44758c8c29..adc663c7e9c895f8d7f5b4741be4cdbfcf9257f0 100644
--- a/index.js
+++ b/index.js
@@ -16,7 +16,7 @@ function transformAndValidate(classType, somethingToTransform, options) {
         else {
             return reject(new Error("Incorrect object param type! Only string, plain object and array of plain objects are valid."));
         }
-        const classObject = class_transformer_1.plainToClass(classType, object, options ? options.transformer : void 0);
+        const classObject = class_transformer_1.plainToInstance(classType, object, options ? options.transformer : void 0);
         if (Array.isArray(classObject)) {
             Promise.all(classObject.map(objectElement => class_validator_1.validate(objectElement, options ? options.validator : void 0))).then(errors => errors.every(error => error.length === 0)
                 ? resolve(classObject)
@@ -42,7 +42,7 @@ function transformAndValidateSync(classType, somethingToTransform, options) {
     else {
         throw new Error("Incorrect object param type! Only string, plain object and array of plain objects are valid.");
     }
-    const classObject = class_transformer_1.plainToClass(classType, object, options ? options.transformer : void 0);
+    const classObject = class_transformer_1.plainToInstance(classType, object, options ? options.transformer : void 0);
     if (Array.isArray(classObject)) {
         const errorsArray = classObject.map(objectElement => class_validator_1.validateSync(objectElement, options ? options.validator : void 0));
         if (errorsArray.some(errors => errors.length !== 0)) {