diff --git a/electron.d.ts b/electron.d.ts
index 87879e2d84763384205807bebb7a935c30d0ca79..230861e5b3e6cdb399416329a0cf53a651dec6ef 100644
--- a/electron.d.ts
+++ b/electron.d.ts
@@ -25735,7 +25735,7 @@ declare namespace NodeJS {
      * `stderr`. Setting this to `true` will silence deprecation warnings. This
      * property is used instead of the `--no-deprecation` command line flag.
      */
-    noDeprecation: boolean;
+    noDeprecation?: boolean;
     /**
      * A `Electron.ParentPort` property if this is a `UtilityProcess` (or `null`
      * otherwise) allowing communication with the parent process.
