diff --git a/binding.gyp b/binding.gyp
index 9be3dcb490b10937416a72a60e1d4290f9c04519..aa070e53809c44d739b44d1771bccc4b0196190d 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -12,7 +12,11 @@
         'MACOSX_DEPLOYMENT_TARGET': '10.7',
       },
       'msvs_settings': {
-        'VCCLCompilerTool': { 'ExceptionHandling': 1 },
+        'VCCLCompilerTool': {
+          'ExceptionHandling': 1,
+          # ARM_BUILD_TWEAK: use default toolset on ARM64, fall back to v142 otherwise
+          'PlatformToolset': '<!(node -p "process.arch === \'arm64\' ? \'\' : \'v142\'")'
+        }
       },
       'include_dirs': ["<!(node -p \"require('node-addon-api').include_dir\")"],
       'sources': [
