diff --git a/binding.gyp b/binding.gyp
index d9c3491906735ff87a581351929e748923388c35..fa5726685909653b04776b942ab7c6a73893739a 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -23,6 +23,18 @@
       "sources": [
         "src/extract.cpp",
       ],
+"msvs_settings": {
+  "VCCLCompilerTool": {
+    "AdditionalOptions": [
+      "-std:c++17",
+    ],
+  }
+},
+"msbuild_settings": {
+  "ClCompile": {
+    "LanguageStandard": "stdcpp17"
+  }
+},
       "defines": [ "NAPI_DISABLE_CPP_EXCEPTIONS" ],
       "conditions": [
         ["OS=='linux'", {
diff --git a/index.js b/index.js
index 967188a6f508e93811b1fa92efc1afd2d5eb7375..6dc29570a61542e8089d4e4dd5633f75ba1c9ee7 100644
--- a/index.js
+++ b/index.js
@@ -1 +1 @@
-module.exports = require('node-gyp-build-optional-packages')(__dirname);
\ No newline at end of file
+module.exports = require('./build/Release/msgpackr-extract.node')