diff --git a/binding.gyp b/binding.gyp
index 7ff7a52602e3d6a3d2cd2a2d4ed10e0268358eb5..434fef172f567c5e1011c33c060abc27cc228ee7 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -23,6 +23,18 @@
       "sources": [
         "src/extract.cpp",
       ],
+    "msvs_settings": {
+      "VCCLCompilerTool": {
+        "AdditionalOptions": [
+          "-std:c17",
+        ],
+      }
+    },
+    "msbuild_settings": {
+      "ClCompile": {
+        "LanguageStandard": "stdcpp17"
+      }
+    },
       "defines": [ "NAPI_DISABLE_CPP_EXCEPTIONS" ],
       "conditions": [
         ["OS=='linux'", {
diff --git a/index.js b/index.js
index 967188a6f508e93811b1fa92efc1afd2d5eb7375..6dc29570a61542e8089d4e4dd5633f75ba1c9ee7 100644
--- a/index.js
+++ b/index.js
@@ -1 +1 @@
-module.exports = require('node-gyp-build-optional-packages')(__dirname);
\ No newline at end of file
+module.exports = require('./build/Release/msgpackr-extract.node')