diff --git a/exec.js b/exec.js
index 3439de5708ddc403e9d92b11b2c407a93cd97b2a..2a30c93e822f79b57ce68af62f562fa98ddf7ec1 100644
--- a/exec.js
+++ b/exec.js
@@ -1,7 +1,7 @@
 // Mini wrapper around `child_process` to make it behave a little like `execa`.
 
-import {promisify} from 'node:util';
-import childProcess from 'node:child_process';
+import {promisify} from 'util';
+import childProcess from 'child_process';
 
 const execFile = promisify(childProcess.execFile);
 