diff --git a/dist/commonjs/guards.d.ts b/dist/commonjs/guards.d.ts
index dcaffcf0efaef357c91b2a6f7411266f38bda57f..4020b549084848d0be844c1461257fa64a601290 100644
--- a/dist/commonjs/guards.d.ts
+++ b/dist/commonjs/guards.d.ts
@@ -51,20 +51,6 @@ declare type DefinitelyBoolean<T> = Extract<T, boolean> extends never ? boolean
  * @category Guard
  */
 export declare function isBoolean<T>(data: T | boolean): data is DefinitelyBoolean<T>;
-declare type DefinitelyPromise<T extends unknown> = Extract<T, Promise<any>> extends never ? Promise<unknown> : Extract<T, Promise<any>>;
-/**
- * A function that checks if the passed parameter is a Promise and narrows it's type accordingly
- * @param data the variable to check
- * @signature
- *    R.isPromise(data)
- * @returns true if the passed input is a Promise, false otherwise
- * @example
- *    R.isPromise(Promise.resolve(5)) //=> true
- *    R.isPromise(Promise.reject(5)) //=> true
- *    R.isPromise('somethingElse') //=> false
- * @category Guard
- */
-export declare function isPromise<T>(data: T): data is DefinitelyPromise<T>;
 declare type DefinitelyArray<T extends unknown> = Extract<T, Array<any> | ReadonlyArray<any>> extends never ? ReadonlyArray<unknown> : Extract<T, Array<any> | ReadonlyArray<any>>;
 /**
  * A function that checks if the passed parameter is an Array and narrows it's type accordingly
diff --git a/dist/es/guards.d.ts b/dist/es/guards.d.ts
index dcaffcf0efaef357c91b2a6f7411266f38bda57f..4020b549084848d0be844c1461257fa64a601290 100644
--- a/dist/es/guards.d.ts
+++ b/dist/es/guards.d.ts
@@ -51,20 +51,6 @@ declare type DefinitelyBoolean<T> = Extract<T, boolean> extends never ? boolean
  * @category Guard
  */
 export declare function isBoolean<T>(data: T | boolean): data is DefinitelyBoolean<T>;
-declare type DefinitelyPromise<T extends unknown> = Extract<T, Promise<any>> extends never ? Promise<unknown> : Extract<T, Promise<any>>;
-/**
- * A function that checks if the passed parameter is a Promise and narrows it's type accordingly
- * @param data the variable to check
- * @signature
- *    R.isPromise(data)
- * @returns true if the passed input is a Promise, false otherwise
- * @example
- *    R.isPromise(Promise.resolve(5)) //=> true
- *    R.isPromise(Promise.reject(5)) //=> true
- *    R.isPromise('somethingElse') //=> false
- * @category Guard
- */
-export declare function isPromise<T>(data: T): data is DefinitelyPromise<T>;
 declare type DefinitelyArray<T extends unknown> = Extract<T, Array<any> | ReadonlyArray<any>> extends never ? ReadonlyArray<unknown> : Extract<T, Array<any> | ReadonlyArray<any>>;
 /**
  * A function that checks if the passed parameter is an Array and narrows it's type accordingly