diff --git a/dist/visitor.js b/dist/visitor.js
index 4fb647535a227ebe25fff6240f2111c1349c6596..6e2a314e1315cc5be7e6af908fae372165ffcf85 100644
--- a/dist/visitor.js
+++ b/dist/visitor.js
@@ -86,7 +86,7 @@ function nodeVisitor(node) {
                     return undefined; // No imports left, elide entire declaration
                 importClause = updatedImportClause;
             }
-            return factory.updateImportDeclaration(node, node.decorators, node.modifiers, importClause, p);
+            return factory.updateImportDeclaration(node, node.decorators, node.modifiers, importClause, p, node.assertClause);
         });
     /**
      * Update ExportDeclaration